// react-photo-editor/dist/react-photo-editor.es.js
import We, { useRef as cr, useState as _, useEffect as Le } from "react";
var je = { exports: {} }, se = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ae;
function fr() {
  if (Ae)
    return se;
  Ae = 1;
  var k = We, P = Symbol.for("react.element"), J = Symbol.for("react.fragment"), S = Object.prototype.hasOwnProperty, M = k.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, $ = { key: !0, ref: !0, __self: !0, __source: !0 };
  function F(j, d, w) {
    var h, b = {}, R = null, W = null;
    w !== void 0 && (R = "" + w), d.key !== void 0 && (R = "" + d.key), d.ref !== void 0 && (W = d.ref);
    for (h in d)
      S.call(d, h) && !$.hasOwnProperty(h) && (b[h] = d[h]);
    if (j && j.defaultProps)
      for (h in d = j.defaultProps, d)
        b[h] === void 0 && (b[h] = d[h]);
    return { $$typeof: P, type: j, key: R, ref: W, props: b, _owner: M.current };
  }
  return se.Fragment = J, se.jsx = F, se.jsxs = F, se;
}
var oe = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Me;
function dr() {
  return Me || (Me = 1, process.env.NODE_ENV !== "production" && function() {
    var k = We, P = Symbol.for("react.element"), J = Symbol.for("react.portal"), S = Symbol.for("react.fragment"), M = Symbol.for("react.strict_mode"), $ = Symbol.for("react.profiler"), F = Symbol.for("react.provider"), j = Symbol.for("react.context"), d = Symbol.for("react.forward_ref"), w = Symbol.for("react.suspense"), h = Symbol.for("react.suspense_list"), b = Symbol.for("react.memo"), R = Symbol.for("react.lazy"), W = Symbol.for("react.offscreen"), I = Symbol.iterator, X = "@@iterator";
    function V(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = I && e[I] || e[X];
      return typeof r == "function" ? r : null;
    }
    var E = k.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function v(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        K("error", e, t);
      }
    }
    function K(e, r, t) {
      {
        var n = E.ReactDebugCurrentFrame, l = n.getStackAddendum();
        l !== "" && (r += "%s", t = t.concat([l]));
        var u = t.map(function(i) {
          return String(i);
        });
        u.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, u);
      }
    }
    var U = !1, Z = !1, D = !1, ie = !1, Q = !1, ee;
    ee = Symbol.for("react.module.reference");
    function re(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === S || e === $ || Q || e === M || e === w || e === h || ie || e === W || U || Z || D || typeof e == "object" && e !== null && (e.$$typeof === R || e.$$typeof === b || e.$$typeof === F || e.$$typeof === j || e.$$typeof === d || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === ee || e.getModuleId !== void 0));
    }
    function le(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var l = r.displayName || r.name || "";
      return l !== "" ? t + "(" + l + ")" : t;
    }
    function N(e) {
      return e.displayName || "Context";
    }
    function p(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && v("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case S:
          return "Fragment";
        case J:
          return "Portal";
        case $:
          return "Profiler";
        case M:
          return "StrictMode";
        case w:
          return "Suspense";
        case h:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case j:
            var r = e;
            return N(r) + ".Consumer";
          case F:
            var t = e;
            return N(t._context) + ".Provider";
          case d:
            return le(e, e.render, "ForwardRef");
          case b:
            var n = e.displayName || null;
            return n !== null ? n : p(e.type) || "Memo";
          case R: {
            var l = e, u = l._payload, i = l._init;
            try {
              return p(i(u));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var C = Object.assign, L = 0, te, ue, ce, Y, fe, de, s;
    function c() {
    }
    c.__reactDisabledLog = !0;
    function x() {
      {
        if (L === 0) {
          te = console.log, ue = console.info, ce = console.warn, Y = console.error, fe = console.group, de = console.groupCollapsed, s = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: c,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        L++;
      }
    }
    function z() {
      {
        if (L--, L === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: C({}, e, {
              value: te
            }),
            info: C({}, e, {
              value: ue
            }),
            warn: C({}, e, {
              value: ce
            }),
            error: C({}, e, {
              value: Y
            }),
            group: C({}, e, {
              value: fe
            }),
            groupCollapsed: C({}, e, {
              value: de
            }),
            groupEnd: C({}, e, {
              value: s
            })
          });
        }
        L < 0 && v("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var T = E.ReactCurrentDispatcher, ae;
    function B(e, r, t) {
      {
        if (ae === void 0)
          try {
            throw Error();
          } catch (l) {
            var n = l.stack.trim().match(/\n( *(at )?)/);
            ae = n && n[1] || "";
          }
        return `
` + ae + e;
      }
    }
    var H = !1, A;
    {
      var Ve = typeof WeakMap == "function" ? WeakMap : Map;
      A = new Ve();
    }
    function we(e, r) {
      if (!e || H)
        return "";
      {
        var t = A.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      H = !0;
      var l = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var u;
      u = T.current, T.current = null, x();
      try {
        if (r) {
          var i = function() {
            throw Error();
          };
          if (Object.defineProperty(i.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(i, []);
            } catch (O) {
              n = O;
            }
            Reflect.construct(e, [], i);
          } else {
            try {
              i.call();
            } catch (O) {
              n = O;
            }
            e.call(i.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (O) {
            n = O;
          }
          e();
        }
      } catch (O) {
        if (O && n && typeof O.stack == "string") {
          for (var o = O.stack.split(`
`), g = n.stack.split(`
`), f = o.length - 1, m = g.length - 1; f >= 1 && m >= 0 && o[f] !== g[m]; )
            m--;
          for (; f >= 1 && m >= 0; f--, m--)
            if (o[f] !== g[m]) {
              if (f !== 1 || m !== 1)
                do
                  if (f--, m--, m < 0 || o[f] !== g[m]) {
                    var y = `
` + o[f].replace(" at new ", " at ");
                    return e.displayName && y.includes("<anonymous>") && (y = y.replace("<anonymous>", e.displayName)), typeof e == "function" && A.set(e, y), y;
                  }
                while (f >= 1 && m >= 0);
              break;
            }
        }
      } finally {
        H = !1, T.current = u, z(), Error.prepareStackTrace = l;
      }
      var G = e ? e.displayName || e.name : "", De = G ? B(G) : "";
      return typeof e == "function" && A.set(e, De), De;
    }
    function Ue(e, r, t) {
      return we(e, !1);
    }
    function Ye(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function ve(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return we(e, Ye(e));
      if (typeof e == "string")
        return B(e);
      switch (e) {
        case w:
          return B("Suspense");
        case h:
          return B("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case d:
            return Ue(e.render);
          case b:
            return ve(e.type, r, t);
          case R: {
            var n = e, l = n._payload, u = n._init;
            try {
              return ve(u(l), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var me = Object.prototype.hasOwnProperty, Re = {}, Ee = E.ReactDebugCurrentFrame;
    function he(e) {
      if (e) {
        var r = e._owner, t = ve(e.type, e._source, r ? r.type : null);
        Ee.setExtraStackFrame(t);
      } else
        Ee.setExtraStackFrame(null);
    }
    function ze(e, r, t, n, l) {
      {
        var u = Function.call.bind(me);
        for (var i in e)
          if (u(e, i)) {
            var o = void 0;
            try {
              if (typeof e[i] != "function") {
                var g = Error((n || "React class") + ": " + t + " type `" + i + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[i] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw g.name = "Invariant Violation", g;
              }
              o = e[i](r, i, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (f) {
              o = f;
            }
            o && !(o instanceof Error) && (he(l), v("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, i, typeof o), he(null)), o instanceof Error && !(o.message in Re) && (Re[o.message] = !0, he(l), v("Failed %s type: %s", t, o.message), he(null));
          }
      }
    }
    var Be = Array.isArray;
    function ge(e) {
      return Be(e);
    }
    function He(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function qe(e) {
      try {
        return _e(e), !1;
      } catch {
        return !0;
      }
    }
    function _e(e) {
      return "" + e;
    }
    function ke(e) {
      if (qe(e))
        return v("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", He(e)), _e(e);
    }
    var ne = E.ReactCurrentOwner, Ge = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Ce, Te, pe;
    pe = {};
    function Je(e) {
      if (me.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Xe(e) {
      if (me.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function Ke(e, r) {
      if (typeof e.ref == "string" && ne.current && r && ne.current.stateNode !== r) {
        var t = p(ne.current.type);
        pe[t] || (v('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', p(ne.current.type), e.ref), pe[t] = !0);
      }
    }
    function Ze(e, r) {
      {
        var t = function() {
          Ce || (Ce = !0, v("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function Qe(e, r) {
      {
        var t = function() {
          Te || (Te = !0, v("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var er = function(e, r, t, n, l, u, i) {
      var o = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: P,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: i,
        // Record the component responsible for creating this element.
        _owner: u
      };
      return o._store = {}, Object.defineProperty(o._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(o, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(o, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: l
      }), Object.freeze && (Object.freeze(o.props), Object.freeze(o)), o;
    };
    function rr(e, r, t, n, l) {
      {
        var u, i = {}, o = null, g = null;
        t !== void 0 && (ke(t), o = "" + t), Xe(r) && (ke(r.key), o = "" + r.key), Je(r) && (g = r.ref, Ke(r, l));
        for (u in r)
          me.call(r, u) && !Ge.hasOwnProperty(u) && (i[u] = r[u]);
        if (e && e.defaultProps) {
          var f = e.defaultProps;
          for (u in f)
            i[u] === void 0 && (i[u] = f[u]);
        }
        if (o || g) {
          var m = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          o && Ze(i, m), g && Qe(i, m);
        }
        return er(e, o, g, l, n, ne.current, i);
      }
    }
    var xe = E.ReactCurrentOwner, Oe = E.ReactDebugCurrentFrame;
    function q(e) {
      if (e) {
        var r = e._owner, t = ve(e.type, e._source, r ? r.type : null);
        Oe.setExtraStackFrame(t);
      } else
        Oe.setExtraStackFrame(null);
    }
    var ye;
    ye = !1;
    function be(e) {
      return typeof e == "object" && e !== null && e.$$typeof === P;
    }
    function Se() {
      {
        if (xe.current) {
          var e = p(xe.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function tr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var Ne = {};
    function ar(e) {
      {
        var r = Se();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function Pe(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = ar(r);
        if (Ne[t])
          return;
        Ne[t] = !0;
        var n = "";
        e && e._owner && e._owner !== xe.current && (n = " It was passed a child from " + p(e._owner.type) + "."), q(e), v('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), q(null);
      }
    }
    function Fe(e, r) {
      {
        if (typeof e != "object")
          return;
        if (ge(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            be(n) && Pe(n, r);
          }
        else if (be(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var l = V(e);
          if (typeof l == "function" && l !== e.entries)
            for (var u = l.call(e), i; !(i = u.next()).done; )
              be(i.value) && Pe(i.value, r);
        }
      }
    }
    function nr(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === d || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === b))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = p(r);
          ze(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !ye) {
          ye = !0;
          var l = p(r);
          v("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", l || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && v("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function sr(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            q(e), v("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), q(null);
            break;
          }
        }
        e.ref !== null && (q(e), v("Invalid attribute `ref` supplied to `React.Fragment`."), q(null));
      }
    }
    function Ie(e, r, t, n, l, u) {
      {
        var i = re(e);
        if (!i) {
          var o = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (o += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var g = tr(l);
          g ? o += g : o += Se();
          var f;
          e === null ? f = "null" : ge(e) ? f = "array" : e !== void 0 && e.$$typeof === P ? (f = "<" + (p(e.type) || "Unknown") + " />", o = " Did you accidentally export a JSX literal instead of a component?") : f = typeof e, v("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", f, o);
        }
        var m = rr(e, r, t, l, u);
        if (m == null)
          return m;
        if (i) {
          var y = r.children;
          if (y !== void 0)
            if (n)
              if (ge(y)) {
                for (var G = 0; G < y.length; G++)
                  Fe(y[G], e);
                Object.freeze && Object.freeze(y);
              } else
                v("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Fe(y, e);
        }
        return e === S ? sr(m) : nr(m), m;
      }
    }
    function or(e, r, t) {
      return Ie(e, r, t, !0);
    }
    function ir(e, r, t) {
      return Ie(e, r, t, !1);
    }
    var lr = ir, ur = or;
    oe.Fragment = S, oe.jsx = lr, oe.jsxs = ur;
  }()), oe;
}
process.env.NODE_ENV === "production" ? je.exports = fr() : je.exports = dr();
var a = je.exports;
const $e = "ml-2 text-md outline-none py-1 px-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700", mr = ({
  file: k,
  onSaveImage: P,
  allowColorEditing: J = !0,
  allowFlip: S = !0,
  allowRotate: M = !0,
  allowZoom: $ = !0,
  downloadOnSave: F,
  open: j,
  onClose: d
}) => {
  const w = cr(null), [h, b] = _(""), [R, W] = _(""), [I, X] = _(100), [V, E] = _(100), [v, K] = _(100), [U, Z] = _(0), [D, ie] = _(0), [Q, ee] = _(!1), [re, le] = _(!1), [N, p] = _(1);
  Le(() => {
    if (k) {
      const s = URL.createObjectURL(k);
      return b(s), W(k.name), () => {
        URL.revokeObjectURL(s);
      };
    }
  }, [k, j]), Le(() => {
    C();
  }, [k, h, D, Q, re, N, I, V, v, U]);
  const C = () => {
    const s = w.current, c = s == null ? void 0 : s.getContext("2d"), x = new Image();
    x.src = h, x.onload = () => {
      if (s && c) {
        const z = x.width * N, T = x.height * N, ae = (x.width - z) / 2, B = (x.height - T) / 2;
        if (s.width = x.width, s.height = x.height, c.filter = L(), c.save(), D) {
          const H = s.width / 2, A = s.height / 2;
          c.translate(H, A), c.rotate(D * Math.PI / 180), c.translate(-H, -A);
        }
        Q && (c.translate(s.width, 0), c.scale(-1, 1)), re && (c.translate(0, s.height), c.scale(1, -1)), c.translate(ae, B), c.scale(N, N), c.drawImage(x, 0, 0, s.width, s.height), c.restore();
      }
    };
  }, L = () => `brightness(${I}%) contrast(${V}%) grayscale(${U}%) saturate(${v}%)`, te = (s) => {
    var c;
    ie(parseInt((c = s == null ? void 0 : s.target) == null ? void 0 : c.value));
  }, ue = () => {
    p((s) => s + 0.1);
  }, ce = () => {
    p((s) => Math.max(s - 0.1, 0.1));
  }, Y = () => {
    X(100), E(100), K(100), Z(0), ie(0), ee(!1), le(!1), p(1);
  }, fe = () => {
    const s = w.current;
    s && (s.toBlob((c) => {
      if (c) {
        const x = new File([c], R, { type: c.type });
        if (F) {
          const z = URL.createObjectURL(c), T = document.createElement("a");
          T.download = `${R}`, T.href = z, T.click(), URL.revokeObjectURL(z);
        }
        P(x), d && d();
      }
    }), Y());
  }, de = () => {
    Y(), d && d();
  };
  return /* @__PURE__ */ a.jsx(a.Fragment, { children: j && /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
    /* @__PURE__ */ a.jsx("div", { "data-testid": "photo-editor-main", className: "photo-editor-main justify-center items-center flex overflow-auto fixed inset-0 z-50", children: /* @__PURE__ */ a.jsxs("div", { className: "relative rounded-lg shadow-lg w-[40rem] max-sm:w-[22rem] bg-white h-[38rem] dark:bg-slate-800", children: [
      /* @__PURE__ */ a.jsxs("div", { className: "flex justify-end p-2 rounded-t", children: [
        /* @__PURE__ */ a.jsx(
          "button",
          {
            className: $e,
            onClick: de,
            children: "Close"
          }
        ),
        /* @__PURE__ */ a.jsx(
          "button",
          {
            className: $e,
            onClick: () => fe(),
            "data-testid": "save-button",
            children: "Save"
          }
        )
      ] }),
      /* @__PURE__ */ a.jsx("div", { className: "p-4", children: /* @__PURE__ */ a.jsxs("div", { className: "flex flex-col", children: [
        /* @__PURE__ */ a.jsx("canvas", { className: "canvas max-w-lg max-h-[22rem] w-full object-fit mx-auto", "data-testid": "image-editor-canvas", id: "canvas", ref: w }),
        /* @__PURE__ */ a.jsx("div", { className: "items-center flex m-1 flex-col", children: /* @__PURE__ */ a.jsxs("div", { className: "flex flex-col bottom-12 gap-1 mt-4 max-sm:w-72 w-11/12 absolute ", children: [
          M && /* @__PURE__ */ a.jsxs("div", { className: "flex flex-row items-center", children: [
            /* @__PURE__ */ a.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Rotate: " }),
            /* @__PURE__ */ a.jsx(
              "input",
              {
                type: "range",
                value: D,
                step: "1",
                onChange: te,
                min: -180,
                max: 180,
                className: "ml-[1.8rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
              }
            ),
            /* @__PURE__ */ a.jsx(
              "input",
              {
                type: "number",
                value: D,
                onChange: te,
                className: "w-14 pl-1 rounded-md text-right"
              }
            )
          ] }),
          J && /* @__PURE__ */ a.jsxs(a.Fragment, { children: [
            /* @__PURE__ */ a.jsxs("div", { className: "flex flex-row justify-between items-center", "data-testid": "color-editing", children: [
              /* @__PURE__ */ a.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Brightness: " }),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  id: "default-range",
                  type: "range",
                  step: "1",
                  max: 200,
                  value: I,
                  onChange: (s) => X(parseInt(s.target.value)),
                  className: "ml-2 w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  type: "number",
                  value: I,
                  onChange: (s) => X(parseInt(s.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] }),
            /* @__PURE__ */ a.jsxs("div", { className: "flex items-center", children: [
              /* @__PURE__ */ a.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Contrast: " }),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  id: "default-range",
                  type: "range",
                  step: "1",
                  max: 200,
                  value: V,
                  onChange: (s) => E(parseInt(s.target.value)),
                  className: "ml-[1.1rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  type: "number",
                  value: V,
                  onChange: (s) => E(parseInt(s.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] }),
            /* @__PURE__ */ a.jsxs("div", { className: "flex items-center", children: [
              /* @__PURE__ */ a.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Saturate: " }),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  id: "default-range",
                  type: "range",
                  step: "1",
                  max: 200,
                  value: v,
                  onChange: (s) => K(parseInt(s.target.value)),
                  className: "ml-[1.2rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  type: "number",
                  value: v,
                  onChange: (s) => K(parseInt(s.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] }),
            /* @__PURE__ */ a.jsxs("div", { className: "flex items-center", children: [
              /* @__PURE__ */ a.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Grayscale: " }),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  id: "grayscaleSlider",
                  type: "range",
                  value: U,
                  max: "100",
                  onChange: (s) => Z(parseInt(s.target.value)),
                  className: "ml-[0.8rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ a.jsx(
                "input",
                {
                  type: "number",
                  value: U,
                  onChange: (s) => Z(parseInt(s.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] })
          ] })
        ] }) }),
        /* @__PURE__ */ a.jsx("div", { className: "flex justify-center", children: /* @__PURE__ */ a.jsxs("div", { className: "mb-1 absolute bottom-0 mt-2", children: [
          /* @__PURE__ */ a.jsx("button", { className: "mx-1  focus:ring-4 rounded-md p-1", onClick: Y, children: /* @__PURE__ */ a.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round", className: "lucide lucide-rotate-ccw dark:stroke-slate-200", children: [
            /* @__PURE__ */ a.jsx("path", { d: "M3 12a9 9 0 1 0 9-9 9.75 9.75 0 0 0-6.74 2.74L3 8" }),
            /* @__PURE__ */ a.jsx("path", { d: "M3 3v5h5" })
          ] }) }),
          S && /* @__PURE__ */ a.jsxs("div", { className: "inline-block", "data-testid": "flip-btns", children: [
            /* @__PURE__ */ a.jsx("button", { className: "mx-1 focus:ring-4 rounded-md p-1", onClick: () => ee(!Q), children: /* @__PURE__ */ a.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-flip-horizontal dark:stroke-slate-200", children: [
              /* @__PURE__ */ a.jsx("path", { d: "M8 3H5a2 2 0 0 0-2 2v14c0 1.1.9 2 2 2h3" }),
              /* @__PURE__ */ a.jsx("path", { d: "M16 3h3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-3" }),
              /* @__PURE__ */ a.jsx("path", { d: "M12 20v2" }),
              /* @__PURE__ */ a.jsx("path", { d: "M12 14v2" }),
              /* @__PURE__ */ a.jsx("path", { d: "M12 8v2" }),
              /* @__PURE__ */ a.jsx("path", { d: "M12 2v2" })
            ] }) }),
            /* @__PURE__ */ a.jsx("button", { className: "mx-1 focus:ring-4 rounded-md p-1", onClick: () => le(!re), children: /* @__PURE__ */ a.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-flip-vertical dark:stroke-slate-200", children: [
              /* @__PURE__ */ a.jsx("path", { d: "M21 8V5a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v3" }),
              /* @__PURE__ */ a.jsx("path", { d: "M21 16v3a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-3" }),
              /* @__PURE__ */ a.jsx("path", { d: "M4 12H2" }),
              /* @__PURE__ */ a.jsx("path", { d: "M10 12H8" }),
              /* @__PURE__ */ a.jsx("path", { d: "M16 12h-2" }),
              /* @__PURE__ */ a.jsx("path", { d: "M22 12h-2" })
            ] }) })
          ] }),
          $ && /* @__PURE__ */ a.jsxs("div", { className: "inline-block", "data-testid": "zoom-btns", children: [
            /* @__PURE__ */ a.jsx("button", { "data-testid": "zoom-in", className: "mx-1 focus:ring-4 rounded-md p-1", onClick: ue, children: /* @__PURE__ */ a.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-zoom-in dark:stroke-slate-200", children: [
              /* @__PURE__ */ a.jsx("circle", { cx: "11", cy: "11", r: "8" }),
              /* @__PURE__ */ a.jsx("line", { x1: "21", x2: "16.65", y1: "21", y2: "16.65" }),
              /* @__PURE__ */ a.jsx("line", { x1: "11", x2: "11", y1: "8", y2: "14" }),
              /* @__PURE__ */ a.jsx("line", { x1: "8", x2: "14", y1: "11", y2: "11" })
            ] }) }),
            /* @__PURE__ */ a.jsx("button", { "data-testid": "zoom-out", className: "mx-1 focus:ring-4 rounded-md p-1", onClick: ce, children: /* @__PURE__ */ a.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-zoom-out dark:stroke-slate-200", children: [
              /* @__PURE__ */ a.jsx("circle", { cx: "11", cy: "11", r: "8" }),
              /* @__PURE__ */ a.jsx("line", { x1: "21", x2: "16.65", y1: "21", y2: "16.65" }),
              /* @__PURE__ */ a.jsx("line", { x1: "8", x2: "14", y1: "11", y2: "11" })
            ] }) })
          ] })
        ] }) })
      ] }) })
    ] }) }),
    /* @__PURE__ */ a.jsx("div", { className: "opacity-75 fixed inset-0 z-40 bg-black" })
  ] }) });
};
export {
  mr as ReactPhotoEditor
};
