// react-files-preview/dist/react-files-preview.es.js
import Re, { createContext as h0, useReducer as v0, useState as He, useEffect as G, useContext as S, Suspense as m0 } from "react";
var _e = { exports: {} }, Q = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ve;
function p0() {
  if (Ve)
    return Q;
  Ve = 1;
  var a = Re, o = Symbol.for("react.element"), p = Symbol.for("react.fragment"), w = Object.prototype.hasOwnProperty, b = a.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, s = { key: !0, ref: !0, __self: !0, __source: !0 };
  function m(x, d, N) {
    var g, T = {}, k = null, V = null;
    N !== void 0 && (k = "" + N), d.key !== void 0 && (k = "" + d.key), d.ref !== void 0 && (V = d.ref);
    for (g in d)
      w.call(d, g) && !s.hasOwnProperty(g) && (T[g] = d[g]);
    if (x && x.defaultProps)
      for (g in d = x.defaultProps, d)
        T[g] === void 0 && (T[g] = d[g]);
    return { $$typeof: o, type: x, key: k, ref: V, props: T, _owner: b.current };
  }
  return Q.Fragment = p, Q.jsx = m, Q.jsxs = m, Q;
}
var ee = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var $e;
function g0() {
  return $e || ($e = 1, process.env.NODE_ENV !== "production" && function() {
    var a = Re, o = Symbol.for("react.element"), p = Symbol.for("react.portal"), w = Symbol.for("react.fragment"), b = Symbol.for("react.strict_mode"), s = Symbol.for("react.profiler"), m = Symbol.for("react.provider"), x = Symbol.for("react.context"), d = Symbol.for("react.forward_ref"), N = Symbol.for("react.suspense"), g = Symbol.for("react.suspense_list"), T = Symbol.for("react.memo"), k = Symbol.for("react.lazy"), V = Symbol.for("react.offscreen"), q = Symbol.iterator, te = "@@iterator";
    function I(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = q && e[q] || e[te];
      return typeof r == "function" ? r : null;
    }
    var F = a.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function j(e) {
      {
        for (var r = arguments.length, n = new Array(r > 1 ? r - 1 : 0), i = 1; i < r; i++)
          n[i - 1] = arguments[i];
        z("error", e, n);
      }
    }
    function z(e, r, n) {
      {
        var i = F.ReactDebugCurrentFrame, f = i.getStackAddendum();
        f !== "" && (r += "%s", n = n.concat([f]));
        var h = n.map(function(u) {
          return String(u);
        });
        h.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, h);
      }
    }
    var re = !1, ae = !1, ne = !1, A = !1, pe = !1, ie;
    ie = Symbol.for("react.module.reference");
    function M(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === w || e === s || pe || e === b || e === N || e === g || A || e === V || re || ae || ne || typeof e == "object" && e !== null && (e.$$typeof === k || e.$$typeof === T || e.$$typeof === m || e.$$typeof === x || e.$$typeof === d || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === ie || e.getModuleId !== void 0));
    }
    function P(e, r, n) {
      var i = e.displayName;
      if (i)
        return i;
      var f = r.displayName || r.name || "";
      return f !== "" ? n + "(" + f + ")" : n;
    }
    function U(e) {
      return e.displayName || "Context";
    }
    function C(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && j("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case w:
          return "Fragment";
        case p:
          return "Portal";
        case s:
          return "Profiler";
        case b:
          return "StrictMode";
        case N:
          return "Suspense";
        case g:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case x:
            var r = e;
            return U(r) + ".Consumer";
          case m:
            var n = e;
            return U(n._context) + ".Provider";
          case d:
            return P(e, e.render, "ForwardRef");
          case T:
            var i = e.displayName || null;
            return i !== null ? i : C(e.type) || "Memo";
          case k: {
            var f = e, h = f._payload, u = f._init;
            try {
              return C(u(h));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var D = Object.assign, $ = 0, le, oe, se, ce, ue, c, v;
    function H() {
    }
    H.__reactDisabledLog = !0;
    function J() {
      {
        if ($ === 0) {
          le = console.log, oe = console.info, se = console.warn, ce = console.error, ue = console.group, c = console.groupCollapsed, v = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: H,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        $++;
      }
    }
    function fe() {
      {
        if ($--, $ === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: D({}, e, {
              value: le
            }),
            info: D({}, e, {
              value: oe
            }),
            warn: D({}, e, {
              value: se
            }),
            error: D({}, e, {
              value: ce
            }),
            group: D({}, e, {
              value: ue
            }),
            groupCollapsed: D({}, e, {
              value: c
            }),
            groupEnd: D({}, e, {
              value: v
            })
          });
        }
        $ < 0 && j("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var W = F.ReactCurrentDispatcher, K;
    function Z(e, r, n) {
      {
        if (K === void 0)
          try {
            throw Error();
          } catch (f) {
            var i = f.stack.trim().match(/\n( *(at )?)/);
            K = i && i[1] || "";
          }
        return `
` + K + e;
      }
    }
    var ge = !1, de;
    {
      var We = typeof WeakMap == "function" ? WeakMap : Map;
      de = new We();
    }
    function Se(e, r) {
      if (!e || ge)
        return "";
      {
        var n = de.get(e);
        if (n !== void 0)
          return n;
      }
      var i;
      ge = !0;
      var f = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var h;
      h = W.current, W.current = null, J();
      try {
        if (r) {
          var u = function() {
            throw Error();
          };
          if (Object.defineProperty(u.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(u, []);
            } catch (L) {
              i = L;
            }
            Reflect.construct(e, [], u);
          } else {
            try {
              u.call();
            } catch (L) {
              i = L;
            }
            e.call(u.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (L) {
            i = L;
          }
          e();
        }
      } catch (L) {
        if (L && i && typeof L.stack == "string") {
          for (var l = L.stack.split(`
`), _ = i.stack.split(`
`), y = l.length - 1, E = _.length - 1; y >= 1 && E >= 0 && l[y] !== _[E]; )
            E--;
          for (; y >= 1 && E >= 0; y--, E--)
            if (l[y] !== _[E]) {
              if (y !== 1 || E !== 1)
                do
                  if (y--, E--, E < 0 || l[y] !== _[E]) {
                    var O = `
` + l[y].replace(" at new ", " at ");
                    return e.displayName && O.includes("<anonymous>") && (O = O.replace("<anonymous>", e.displayName)), typeof e == "function" && de.set(e, O), O;
                  }
                while (y >= 1 && E >= 0);
              break;
            }
        }
      } finally {
        ge = !1, W.current = h, fe(), Error.prepareStackTrace = f;
      }
      var Y = e ? e.displayName || e.name : "", Me = Y ? Z(Y) : "";
      return typeof e == "function" && de.set(e, Me), Me;
    }
    function Ze(e, r, n) {
      return Se(e, !1);
    }
    function Be(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function he(e, r, n) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return Se(e, Be(e));
      if (typeof e == "string")
        return Z(e);
      switch (e) {
        case N:
          return Z("Suspense");
        case g:
          return Z("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case d:
            return Ze(e.render);
          case T:
            return he(e.type, r, n);
          case k: {
            var i = e, f = i._payload, h = i._init;
            try {
              return he(h(f), r, n);
            } catch {
            }
          }
        }
      return "";
    }
    var ve = Object.prototype.hasOwnProperty, Te = {}, ke = F.ReactDebugCurrentFrame;
    function me(e) {
      if (e) {
        var r = e._owner, n = he(e.type, e._source, r ? r.type : null);
        ke.setExtraStackFrame(n);
      } else
        ke.setExtraStackFrame(null);
    }
    function Ye(e, r, n, i, f) {
      {
        var h = Function.call.bind(ve);
        for (var u in e)
          if (h(e, u)) {
            var l = void 0;
            try {
              if (typeof e[u] != "function") {
                var _ = Error((i || "React class") + ": " + n + " type `" + u + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[u] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw _.name = "Invariant Violation", _;
              }
              l = e[u](r, u, i, n, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (y) {
              l = y;
            }
            l && !(l instanceof Error) && (me(f), j("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", i || "React class", n, u, typeof l), me(null)), l instanceof Error && !(l.message in Te) && (Te[l.message] = !0, me(f), j("Failed %s type: %s", n, l.message), me(null));
          }
      }
    }
    var Ge = Array.isArray;
    function xe(e) {
      return Ge(e);
    }
    function qe(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, n = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return n;
      }
    }
    function Je(e) {
      try {
        return Ce(e), !1;
      } catch {
        return !0;
      }
    }
    function Ce(e) {
      return "" + e;
    }
    function Oe(e) {
      if (Je(e))
        return j("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", qe(e)), Ce(e);
    }
    var X = F.ReactCurrentOwner, Ke = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Ne, Ae, we;
    we = {};
    function Xe(e) {
      if (ve.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Qe(e) {
      if (ve.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function e0(e, r) {
      if (typeof e.ref == "string" && X.current && r && X.current.stateNode !== r) {
        var n = C(X.current.type);
        we[n] || (j('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', C(X.current.type), e.ref), we[n] = !0);
      }
    }
    function t0(e, r) {
      {
        var n = function() {
          Ne || (Ne = !0, j("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: n,
          configurable: !0
        });
      }
    }
    function r0(e, r) {
      {
        var n = function() {
          Ae || (Ae = !0, j("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        n.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: n,
          configurable: !0
        });
      }
    }
    var a0 = function(e, r, n, i, f, h, u) {
      var l = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: o,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: n,
        props: u,
        // Record the component responsible for creating this element.
        _owner: h
      };
      return l._store = {}, Object.defineProperty(l._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(l, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: i
      }), Object.defineProperty(l, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: f
      }), Object.freeze && (Object.freeze(l.props), Object.freeze(l)), l;
    };
    function n0(e, r, n, i, f) {
      {
        var h, u = {}, l = null, _ = null;
        n !== void 0 && (Oe(n), l = "" + n), Qe(r) && (Oe(r.key), l = "" + r.key), Xe(r) && (_ = r.ref, e0(r, f));
        for (h in r)
          ve.call(r, h) && !Ke.hasOwnProperty(h) && (u[h] = r[h]);
        if (e && e.defaultProps) {
          var y = e.defaultProps;
          for (h in y)
            u[h] === void 0 && (u[h] = y[h]);
        }
        if (l || _) {
          var E = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          l && t0(u, E), _ && r0(u, E);
        }
        return a0(e, l, _, f, i, X.current, u);
      }
    }
    var ye = F.ReactCurrentOwner, Fe = F.ReactDebugCurrentFrame;
    function B(e) {
      if (e) {
        var r = e._owner, n = he(e.type, e._source, r ? r.type : null);
        Fe.setExtraStackFrame(n);
      } else
        Fe.setExtraStackFrame(null);
    }
    var be;
    be = !1;
    function Ee(e) {
      return typeof e == "object" && e !== null && e.$$typeof === o;
    }
    function Pe() {
      {
        if (ye.current) {
          var e = C(ye.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function i0(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), n = e.lineNumber;
          return `

Check your code at ` + r + ":" + n + ".";
        }
        return "";
      }
    }
    var De = {};
    function l0(e) {
      {
        var r = Pe();
        if (!r) {
          var n = typeof e == "string" ? e : e.displayName || e.name;
          n && (r = `

Check the top-level render call using <` + n + ">.");
        }
        return r;
      }
    }
    function Le(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var n = l0(r);
        if (De[n])
          return;
        De[n] = !0;
        var i = "";
        e && e._owner && e._owner !== ye.current && (i = " It was passed a child from " + C(e._owner.type) + "."), B(e), j('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', n, i), B(null);
      }
    }
    function Ie(e, r) {
      {
        if (typeof e != "object")
          return;
        if (xe(e))
          for (var n = 0; n < e.length; n++) {
            var i = e[n];
            Ee(i) && Le(i, r);
          }
        else if (Ee(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var f = I(e);
          if (typeof f == "function" && f !== e.entries)
            for (var h = f.call(e), u; !(u = h.next()).done; )
              Ee(u.value) && Le(u.value, r);
        }
      }
    }
    function o0(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var n;
        if (typeof r == "function")
          n = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === d || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === T))
          n = r.propTypes;
        else
          return;
        if (n) {
          var i = C(r);
          Ye(n, e.props, "prop", i, e);
        } else if (r.PropTypes !== void 0 && !be) {
          be = !0;
          var f = C(r);
          j("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", f || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && j("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function s0(e) {
      {
        for (var r = Object.keys(e.props), n = 0; n < r.length; n++) {
          var i = r[n];
          if (i !== "children" && i !== "key") {
            B(e), j("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", i), B(null);
            break;
          }
        }
        e.ref !== null && (B(e), j("Invalid attribute `ref` supplied to `React.Fragment`."), B(null));
      }
    }
    function ze(e, r, n, i, f, h) {
      {
        var u = M(e);
        if (!u) {
          var l = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (l += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var _ = i0(f);
          _ ? l += _ : l += Pe();
          var y;
          e === null ? y = "null" : xe(e) ? y = "array" : e !== void 0 && e.$$typeof === o ? (y = "<" + (C(e.type) || "Unknown") + " />", l = " Did you accidentally export a JSX literal instead of a component?") : y = typeof e, j("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", y, l);
        }
        var E = n0(e, r, n, f, h);
        if (E == null)
          return E;
        if (u) {
          var O = r.children;
          if (O !== void 0)
            if (i)
              if (xe(O)) {
                for (var Y = 0; Y < O.length; Y++)
                  Ie(O[Y], e);
                Object.freeze && Object.freeze(O);
              } else
                j("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Ie(O, e);
        }
        return e === w ? s0(E) : o0(E), E;
      }
    }
    function c0(e, r, n) {
      return ze(e, r, n, !0);
    }
    function u0(e, r, n) {
      return ze(e, r, n, !1);
    }
    var f0 = u0, d0 = c0;
    ee.Fragment = w, ee.jsx = f0, ee.jsxs = d0;
  }()), ee;
}
process.env.NODE_ENV === "production" ? _e.exports = p0() : _e.exports = g0();
var t = _e.exports;
const x0 = [
  "image/jpeg",
  "image/jpg",
  "image/png",
  "image/gif",
  "image/tiff"
], w0 = (a, o) => {
  const p = a.fileData.length - 1, w = (s) => {
    const m = a.fileData[s];
    return {
      zoom: !0,
      fileSrc: URL.createObjectURL(m),
      index: s,
      isImage: x0.includes(m.type),
      fileName: m.name,
      type: m.type,
      size: m.size
    };
  }, b = (s) => {
    s && s.fileSrc && s.fileSrc.startsWith("blob:") && URL.revokeObjectURL(s.fileSrc);
  };
  switch (o.type) {
    case "STORE_FILE_DATA":
      return { ...a, fileData: o.payload.files };
    case "SET_COMPONENT_STATE":
      return { ...a, componentState: o.payload };
    case "APPEND_FILE_DATA":
      return { ...a, fileData: [...a.fileData, ...o.payload.files] };
    case "STORE_FILE_STATE":
      return { ...a, fileState: o.payload };
    case "SET_IMAGE_EDITOR_DATA":
      return { ...a, imageEditorState: o.payload };
    case "REMOVE_FILE_DATA":
      return {
        ...a,
        fileData: a.fileData.filter((s) => s.name !== o.payload.name)
      };
    case "GET_NEXT_FILE": {
      const s = a.fileState.index + 1, m = s > p ? 0 : s;
      return b(a.fileState), {
        ...a,
        fileState: w(m)
      };
    }
    case "GET_PREV_FILE": {
      const s = a.fileState.index - 1, m = s < 0 ? p : s;
      return b(a.fileState), {
        ...a,
        fileState: w(m)
      };
    }
    default:
      return a;
  }
}, R = h0({
  state: {
    fileData: [],
    fileState: {
      zoom: !1,
      fileSrc: null,
      index: 0,
      isImage: !1,
      fileName: null,
      type: null,
      size: 0
    },
    componentState: {
      showFileSize: !0,
      showSliderCount: !0,
      downloadFile: !0,
      removeFile: !0,
      rounded: !0,
      fileHeight: "h-32",
      fileWidth: "w-44",
      disabled: !1,
      allowEditing: !1
    },
    imageEditorState: {
      isEditing: !1,
      index: null,
      file: null
    }
  },
  dispatch: () => {
  }
}), y0 = ({ children: a }) => {
  const [o, p] = v0(w0, {
    fileData: [],
    fileState: {
      zoom: !1,
      fileSrc: null,
      index: 0,
      isImage: !1,
      fileName: null,
      type: null,
      size: 0
    },
    componentState: {
      showFileSize: !0,
      showSliderCount: !0,
      downloadFile: !0,
      removeFile: !0,
      rounded: !0,
      fileHeight: "h-32",
      fileWidth: "w-44"
    },
    imageEditorState: {
      isEditing: !1,
      index: null,
      file: null
    }
  });
  return /* @__PURE__ */ t.jsx(R.Provider, { value: { state: o, dispatch: p }, children: a });
}, b0 = ({ file: a, fileSrc: o, index: p, isImage: w }) => {
  const [b, s] = He(null);
  G(() => {
    a.size < 1e6 ? s(Math.floor(a.size / 1e3) + " KB") : s(Math.floor(a.size / 1e6) + " MB");
  }, [a]);
  const m = a.name.split(".");
  let x = m[0];
  const d = m.pop();
  x.length > 20 && (x = x.substring(0, 5) + ".." + x.substring(x.length - 3, x.length));
  const N = x + "." + d, g = S(R).state.componentState, { dispatch: T } = S(R), k = () => {
    T({
      type: "SET_IMAGE_EDITOR_DATA",
      payload: {
        isEditing: !0,
        file: a,
        index: p
      }
    });
  };
  return /* @__PURE__ */ t.jsxs("div", { className: "relative ", children: [
    /* @__PURE__ */ t.jsx("h5", { className: "text-[12px] mt-1 font-normal break-words text-black dark:text-slate-100", children: N }),
    g.showFileSize && /* @__PURE__ */ t.jsx("span", { "data-testid": "file-size", className: "text-[10px] text-black dark:text-slate-100", children: b }),
    g.downloadFile && o && /* @__PURE__ */ t.jsx(
      "a",
      {
        className: "float-right absolute top-1 right-0 text-gray-500 dark:text-white",
        href: o,
        target: "_blank",
        download: !0,
        rel: "noreferrer",
        children: /* @__PURE__ */ t.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", className: "h-4 w-4", fill: "currentColor", viewBox: "0 0 16 16", children: [
          /* @__PURE__ */ t.jsx("path", { d: "M.5 9.9a.5.5 0 0 1 .5.5v2.5a1 1 0 0 0 1 1h12a1 1 0 0 0 1-1v-2.5a.5.5 0 0 1 1 0v2.5a2 2 0 0 1-2 2H2a2 2 0 0 1-2-2v-2.5a.5.5 0 0 1 .5-.5z" }),
          /* @__PURE__ */ t.jsx("path", { d: "M7.646 11.854a.5.5 0 0 0 .708 0l3-3a.5.5 0 0 0-.708-.708L8.5 10.293V1.5a.5.5 0 0 0-1 0v8.793L5.354 8.146a.5.5 0 1 0-.708.708l3 3z" })
        ] })
      }
    ),
    g.allowEditing && w && /* @__PURE__ */ t.jsx(
      "button",
      {
        className: "float-right absolute top-1 right-5 text-gray-500 dark:text-white",
        onClick: k,
        children: /* @__PURE__ */ t.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "16", height: "16", fill: "currentColor", className: "h-4 w-4", viewBox: "0 0 16 16", children: [
          /* @__PURE__ */ t.jsx("path", { d: "M15.502 1.94a.5.5 0 0 1 0 .706L14.459 3.69l-2-2L13.502.646a.5.5 0 0 1 .707 0l1.293 1.293zm-1.75 2.456-2-2L4.939 9.21a.5.5 0 0 0-.121.196l-.805 2.414a.25.25 0 0 0 .316.316l2.414-.805a.5.5 0 0 0 .196-.12l6.813-6.814z" }),
          /* @__PURE__ */ t.jsx("path", { fillRule: "evenodd", d: "M1 13.5A1.5 1.5 0 0 0 2.5 15h11a1.5 1.5 0 0 0 1.5-1.5v-6a.5.5 0 0 0-1 0v6a.5.5 0 0 1-.5.5h-11a.5.5 0 0 1-.5-.5v-11a.5.5 0 0 1 .5-.5H9a.5.5 0 0 0 0-1H2.5A1.5 1.5 0 0 0 1 2.5v11z" })
        ] })
      }
    )
  ] });
}, Ue = [
  {
    type: "application/pdf",
    icon: /* @__PURE__ */ t.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", className: "h-10 fill-white", fill: "currentColor", viewBox: "0 0 16 16", children: [
      /* @__PURE__ */ t.jsx("path", { d: "M5.523 10.424c.14-.082.293-.162.459-.238a7.878 7.878 0 0 1-.45.606c-.28.337-.498.516-.635.572a.266.266 0 0 1-.035.012.282.282 0 0 1-.026-.044c-.056-.11-.054-.216.04-.36.106-.165.319-.354.647-.548zm2.455-1.647c-.119.025-.237.05-.356.078a21.035 21.035 0 0 0 .5-1.05 11.96 11.96 0 0 0 .51.858c-.217.032-.436.07-.654.114zm2.525.939a3.888 3.888 0 0 1-.435-.41c.228.005.434.022.612.054.317.057.466.147.518.209a.095.095 0 0 1 .026.064.436.436 0 0 1-.06.2.307.307 0 0 1-.094.124.107.107 0 0 1-.069.015c-.09-.003-.258-.066-.498-.256zM8.278 4.97c-.04.244-.108.524-.2.829a4.86 4.86 0 0 1-.089-.346c-.076-.353-.087-.63-.046-.822.038-.177.11-.248.196-.283a.517.517 0 0 1 .145-.04c.013.03.028.092.032.198.005.122-.007.277-.038.465z" }),
      /* @__PURE__ */ t.jsx("path", { fillRule: "evenodd", d: "M4 0h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2zm.165 11.668c.09.18.23.343.438.419.207.075.412.04.58-.03.318-.13.635-.436.926-.786.333-.401.683-.927 1.021-1.51a11.64 11.64 0 0 1 1.997-.406c.3.383.61.713.91.95.28.22.603.403.934.417a.856.856 0 0 0 .51-.138c.155-.101.27-.247.354-.416.09-.181.145-.37.138-.563a.844.844 0 0 0-.2-.518c-.226-.27-.596-.4-.96-.465a5.76 5.76 0 0 0-1.335-.05 10.954 10.954 0 0 1-.98-1.686c.25-.66.437-1.284.52-1.794.036-.218.055-.426.048-.614a1.238 1.238 0 0 0-.127-.538.7.7 0 0 0-.477-.365c-.202-.043-.41 0-.601.077-.377.15-.576.47-.651.823-.073.34-.04.736.046 1.136.088.406.238.848.43 1.295a19.707 19.707 0 0 1-1.062 2.227 7.662 7.662 0 0 0-1.482.645c-.37.22-.699.48-.897.787-.21.326-.275.714-.08 1.103z" })
    ] }),
    color: "bg-red-500"
  },
  {
    type: "text/csv",
    icon: /* @__PURE__ */ t.jsx("svg", { xmlns: "http://www.w3.org/2000/svg", className: "h-10 fill-white", fill: "currentColor", viewBox: "0 0 16 16", children: /* @__PURE__ */ t.jsx("path", { fillRule: "evenodd", d: "M14 4.5V14a2 2 0 0 1-2 2h-1v-1h1a1 1 0 0 0 1-1V4.5h-2A1.5 1.5 0 0 1 9.5 3V1H4a1 1 0 0 0-1 1v9H2V2a2 2 0 0 1 2-2h5.5L14 4.5ZM3.397 14.841a1.13 1.13 0 0 0 .401.823c.13.108.289.192.478.252.19.061.411.091.665.091.338 0 .624-.053.859-.158.236-.105.416-.252.539-.44.125-.189.187-.408.187-.656 0-.224-.045-.41-.134-.56a1.001 1.001 0 0 0-.375-.357 2.027 2.027 0 0 0-.566-.21l-.621-.144a.97.97 0 0 1-.404-.176.37.37 0 0 1-.144-.299c0-.156.062-.284.185-.384.125-.101.296-.152.512-.152.143 0 .266.023.37.068a.624.624 0 0 1 .246.181.56.56 0 0 1 .12.258h.75a1.092 1.092 0 0 0-.2-.566 1.21 1.21 0 0 0-.5-.41 1.813 1.813 0 0 0-.78-.152c-.293 0-.551.05-.776.15-.225.099-.4.24-.527.421-.127.182-.19.395-.19.639 0 .201.04.376.122.524.082.149.2.27.352.367.152.095.332.167.539.213l.618.144c.207.049.361.113.463.193a.387.387 0 0 1 .152.326.505.505 0 0 1-.085.29.559.559 0 0 1-.255.193c-.111.047-.249.07-.413.07-.117 0-.223-.013-.32-.04a.838.838 0 0 1-.248-.115.578.578 0 0 1-.255-.384h-.765ZM.806 13.693c0-.248.034-.46.102-.633a.868.868 0 0 1 .302-.399.814.814 0 0 1 .475-.137c.15 0 .283.032.398.097a.7.7 0 0 1 .272.26.85.85 0 0 1 .12.381h.765v-.072a1.33 1.33 0 0 0-.466-.964 1.441 1.441 0 0 0-.489-.272 1.838 1.838 0 0 0-.606-.097c-.356 0-.66.074-.911.223-.25.148-.44.359-.572.632-.13.274-.196.6-.196.979v.498c0 .379.064.704.193.976.131.271.322.48.572.626.25.145.554.217.914.217.293 0 .554-.055.785-.164.23-.11.414-.26.55-.454a1.27 1.27 0 0 0 .226-.674v-.076h-.764a.799.799 0 0 1-.118.363.7.7 0 0 1-.272.25.874.874 0 0 1-.401.087.845.845 0 0 1-.478-.132.833.833 0 0 1-.299-.392 1.699 1.699 0 0 1-.102-.627v-.495ZM6.78 15.29a1.176 1.176 0 0 1-.111-.449h.764a.578.578 0 0 0 .255.384c.07.049.154.087.25.114.095.028.201.041.319.041.164 0 .301-.023.413-.07a.559.559 0 0 0 .255-.193.507.507 0 0 0 .085-.29.387.387 0 0 0-.153-.326c-.101-.08-.256-.144-.463-.193l-.618-.143a1.72 1.72 0 0 1-.539-.214 1 1 0 0 1-.351-.367 1.068 1.068 0 0 1-.123-.524c0-.244.063-.457.19-.639.127-.181.303-.322.527-.422.225-.1.484-.149.777-.149.304 0 .564.05.779.152.217.102.384.239.5.41.12.17.187.359.2.566h-.75a.56.56 0 0 0-.12-.258.624.624 0 0 0-.246-.181.923.923 0 0 0-.37-.068c-.216 0-.387.05-.512.152a.472.472 0 0 0-.184.384c0 .121.047.22.143.3a.97.97 0 0 0 .404.175l.621.143c.217.05.406.12.566.211.16.09.285.21.375.358.09.148.135.335.135.56 0 .247-.063.466-.188.656a1.216 1.216 0 0 1-.539.439c-.234.105-.52.158-.858.158-.254 0-.476-.03-.665-.09a1.404 1.404 0 0 1-.478-.252 1.13 1.13 0 0 1-.29-.375Z" }) }),
    color: "bg-emerald-600"
  },
  {
    type: "text/plain",
    icon: /* @__PURE__ */ t.jsxs("svg", { "data-testid": "preview-icon", xmlns: "http://www.w3.org/2000/svg", className: "h-10 fill-white text-4xl", fill: "currentColor", viewBox: "0 0 16 16", children: [
      /* @__PURE__ */ t.jsx("path", { d: "M5 4a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm-.5 2.5A.5.5 0 0 1 5 6h6a.5.5 0 0 1 0 1H5a.5.5 0 0 1-.5-.5zM5 8a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1H5z" }),
      /* @__PURE__ */ t.jsx("path", { d: "M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1z" })
    ] }),
    color: "bg-slate-500"
  },
  {
    type: "application/msword",
    icon: /* @__PURE__ */ t.jsx("svg", { xmlns: "http://www.w3.org/2000/svg", className: "h-10 fill-white", fill: "currentColor", viewBox: "0 0 16 16", children: /* @__PURE__ */ t.jsx("path", { fillRule: "evenodd", d: "M14 4.5V14a2 2 0 0 1-2 2v-1a1 1 0 0 0 1-1V4.5h-2A1.5 1.5 0 0 1 9.5 3V1H4a1 1 0 0 0-1 1v9H2V2a2 2 0 0 1 2-2h5.5L14 4.5Zm-7.839 9.166v.522c0 .256-.039.47-.117.641a.861.861 0 0 1-.322.387.877.877 0 0 1-.469.126.883.883 0 0 1-.471-.126.868.868 0 0 1-.32-.386 1.55 1.55 0 0 1-.117-.642v-.522c0-.257.04-.471.117-.641a.868.868 0 0 1 .32-.387.868.868 0 0 1 .471-.129c.176 0 .332.043.469.13a.861.861 0 0 1 .322.386c.078.17.117.384.117.641Zm.803.519v-.513c0-.377-.068-.7-.205-.972a1.46 1.46 0 0 0-.589-.63c-.254-.147-.56-.22-.917-.22-.355 0-.662.073-.92.22a1.441 1.441 0 0 0-.589.627c-.136.271-.205.596-.205.975v.513c0 .375.069.7.205.973.137.271.333.48.59.627.257.144.564.216.92.216.357 0 .662-.072.916-.216.256-.147.452-.356.59-.627.136-.274.204-.598.204-.973ZM0 11.926v4h1.459c.402 0 .735-.08.999-.238a1.45 1.45 0 0 0 .595-.689c.13-.3.196-.662.196-1.084 0-.42-.065-.778-.196-1.075a1.426 1.426 0 0 0-.59-.68c-.263-.156-.598-.234-1.004-.234H0Zm.791.645h.563c.248 0 .45.05.609.152a.89.89 0 0 1 .354.454c.079.201.118.452.118.753a2.3 2.3 0 0 1-.068.592 1.141 1.141 0 0 1-.196.422.8.8 0 0 1-.334.252 1.298 1.298 0 0 1-.483.082H.79V12.57Zm7.422.483a1.732 1.732 0 0 0-.103.633v.495c0 .246.034.455.103.627a.834.834 0 0 0 .298.393.845.845 0 0 0 .478.131.868.868 0 0 0 .401-.088.699.699 0 0 0 .273-.248.8.8 0 0 0 .117-.364h.765v.076a1.268 1.268 0 0 1-.226.674c-.137.194-.32.345-.55.454a1.81 1.81 0 0 1-.786.164c-.36 0-.664-.072-.914-.216a1.424 1.424 0 0 1-.571-.627c-.13-.272-.194-.597-.194-.976v-.498c0-.379.066-.705.197-.978.13-.274.321-.485.571-.633.252-.149.556-.223.911-.223.219 0 .421.032.607.097.187.062.35.153.489.272a1.326 1.326 0 0 1 .466.964v.073H9.78a.85.85 0 0 0-.12-.38.7.7 0 0 0-.273-.261.802.802 0 0 0-.398-.097.814.814 0 0 0-.475.138.868.868 0 0 0-.301.398Z" }) }),
    color: "bg-sky-600"
  },
  {
    type: "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
    icon: /* @__PURE__ */ t.jsx("svg", { xmlns: "http://www.w3.org/2000/svg", className: "h-10 fill-white", fill: "currentColor", viewBox: "0 0 16 16", children: /* @__PURE__ */ t.jsx("path", { fillRule: "evenodd", d: "M14 4.5V11h-1V4.5h-2A1.5 1.5 0 0 1 9.5 3V1H4a1 1 0 0 0-1 1v9H2V2a2 2 0 0 1 2-2h5.5L14 4.5Zm-6.839 9.688v-.522a1.54 1.54 0 0 0-.117-.641.861.861 0 0 0-.322-.387.862.862 0 0 0-.469-.129.868.868 0 0 0-.471.13.868.868 0 0 0-.32.386 1.54 1.54 0 0 0-.117.641v.522c0 .256.04.47.117.641a.868.868 0 0 0 .32.387.883.883 0 0 0 .471.126.877.877 0 0 0 .469-.126.861.861 0 0 0 .322-.386 1.55 1.55 0 0 0 .117-.642Zm.803-.516v.513c0 .375-.068.7-.205.973a1.47 1.47 0 0 1-.589.627c-.254.144-.56.216-.917.216a1.86 1.86 0 0 1-.92-.216 1.463 1.463 0 0 1-.589-.627 2.151 2.151 0 0 1-.205-.973v-.513c0-.379.069-.704.205-.975.137-.274.333-.483.59-.627.257-.147.564-.22.92-.22.357 0 .662.073.916.22.256.146.452.356.59.63.136.271.204.595.204.972ZM1 15.925v-3.999h1.459c.406 0 .741.078 1.005.235.264.156.46.382.589.68.13.296.196.655.196 1.074 0 .422-.065.784-.196 1.084-.131.301-.33.53-.595.689-.264.158-.597.237-.999.237H1Zm1.354-3.354H1.79v2.707h.563c.185 0 .346-.028.483-.082a.8.8 0 0 0 .334-.252c.088-.114.153-.254.196-.422a2.3 2.3 0 0 0 .068-.592c0-.3-.04-.552-.118-.753a.89.89 0 0 0-.354-.454c-.158-.102-.361-.152-.61-.152Zm6.756 1.116c0-.248.034-.46.103-.633a.868.868 0 0 1 .301-.398.814.814 0 0 1 .475-.138c.15 0 .283.032.398.097a.7.7 0 0 1 .273.26.85.85 0 0 1 .12.381h.765v-.073a1.33 1.33 0 0 0-.466-.964 1.44 1.44 0 0 0-.49-.272 1.836 1.836 0 0 0-.606-.097c-.355 0-.66.074-.911.223-.25.148-.44.359-.571.633-.131.273-.197.6-.197.978v.498c0 .379.065.704.194.976.13.271.321.48.571.627.25.144.555.216.914.216.293 0 .555-.054.785-.164.23-.11.414-.26.551-.454a1.27 1.27 0 0 0 .226-.674v-.076h-.765a.8.8 0 0 1-.117.364.699.699 0 0 1-.273.248.874.874 0 0 1-.401.088.845.845 0 0 1-.478-.131.834.834 0 0 1-.298-.393 1.7 1.7 0 0 1-.103-.627v-.495Zm5.092-1.76h.894l-1.275 2.006 1.254 1.992h-.908l-.85-1.415h-.035l-.852 1.415h-.862l1.24-2.015-1.228-1.984h.932l.832 1.439h.035l.823-1.439Z" }) }),
    color: "bg-sky-600"
  },
  {
    type: "application/vnd.ms-excel",
    icon: /* @__PURE__ */ t.jsx("svg", { xmlns: "http://www.w3.org/2000/svg", className: "h-10 fill-white", fill: "currentColor", viewBox: "0 0 16 16", children: /* @__PURE__ */ t.jsx("path", { fillRule: "evenodd", d: "M14 4.5V14a2 2 0 0 1-2 2h-1v-1h1a1 1 0 0 0 1-1V4.5h-2A1.5 1.5 0 0 1 9.5 3V1H4a1 1 0 0 0-1 1v9H2V2a2 2 0 0 1 2-2h5.5L14 4.5ZM6.472 15.29a1.176 1.176 0 0 1-.111-.449h.765a.578.578 0 0 0 .254.384c.07.049.154.087.25.114.095.028.202.041.319.041.164 0 .302-.023.413-.07a.559.559 0 0 0 .255-.193.507.507 0 0 0 .085-.29.387.387 0 0 0-.153-.326c-.101-.08-.255-.144-.462-.193l-.619-.143a1.72 1.72 0 0 1-.539-.214 1.001 1.001 0 0 1-.351-.367 1.068 1.068 0 0 1-.123-.524c0-.244.063-.457.19-.639.127-.181.303-.322.527-.422.225-.1.484-.149.777-.149.305 0 .564.05.78.152.216.102.383.239.5.41.12.17.186.359.2.566h-.75a.56.56 0 0 0-.12-.258.625.625 0 0 0-.247-.181.923.923 0 0 0-.369-.068c-.217 0-.388.05-.513.152a.472.472 0 0 0-.184.384c0 .121.048.22.143.3a.97.97 0 0 0 .405.175l.62.143c.217.05.406.12.566.211a1 1 0 0 1 .375.358c.09.148.135.335.135.56 0 .247-.063.466-.188.656a1.216 1.216 0 0 1-.539.439c-.234.105-.52.158-.858.158-.254 0-.476-.03-.665-.09a1.404 1.404 0 0 1-.478-.252 1.13 1.13 0 0 1-.29-.375Zm-2.945-3.358h-.893L1.81 13.37h-.036l-.832-1.438h-.93l1.227 1.983L0 15.931h.861l.853-1.415h.035l.85 1.415h.908L2.253 13.94l1.274-2.007Zm2.727 3.325H4.557v-3.325h-.79v4h2.487v-.675Z" }) }),
    color: "bg-emerald-600"
  },
  {
    type: "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
    icon: /* @__PURE__ */ t.jsx("svg", { xmlns: "http://www.w3.org/2000/svg", className: "h-10 fill-white", fill: "currentColor", viewBox: "0 0 16 16", children: /* @__PURE__ */ t.jsx("path", { fillRule: "evenodd", d: "M14 4.5V11h-1V4.5h-2A1.5 1.5 0 0 1 9.5 3V1H4a1 1 0 0 0-1 1v9H2V2a2 2 0 0 1 2-2h5.5L14 4.5ZM7.86 14.841a1.13 1.13 0 0 0 .401.823c.13.108.29.192.479.252.19.061.411.091.665.091.338 0 .624-.053.858-.158.237-.105.416-.252.54-.44a1.17 1.17 0 0 0 .187-.656c0-.224-.045-.41-.135-.56a1.002 1.002 0 0 0-.375-.357 2.028 2.028 0 0 0-.565-.21l-.621-.144a.97.97 0 0 1-.405-.176.37.37 0 0 1-.143-.299c0-.156.061-.284.184-.384.125-.101.296-.152.513-.152.143 0 .266.023.37.068a.624.624 0 0 1 .245.181.56.56 0 0 1 .12.258h.75a1.093 1.093 0 0 0-.199-.566 1.21 1.21 0 0 0-.5-.41 1.813 1.813 0 0 0-.78-.152c-.293 0-.552.05-.777.15-.224.099-.4.24-.527.421-.127.182-.19.395-.19.639 0 .201.04.376.123.524.082.149.199.27.351.367.153.095.332.167.54.213l.618.144c.207.049.36.113.462.193a.387.387 0 0 1 .153.326.512.512 0 0 1-.085.29.558.558 0 0 1-.255.193c-.111.047-.25.07-.413.07-.117 0-.224-.013-.32-.04a.837.837 0 0 1-.249-.115.578.578 0 0 1-.255-.384h-.764Zm-3.726-2.909h.893l-1.274 2.007 1.254 1.992h-.908l-.85-1.415h-.035l-.853 1.415H1.5l1.24-2.016-1.228-1.983h.931l.832 1.438h.036l.823-1.438Zm1.923 3.325h1.697v.674H5.266v-3.999h.791v3.325Zm7.636-3.325h.893l-1.274 2.007 1.254 1.992h-.908l-.85-1.415h-.035l-.853 1.415h-.861l1.24-2.016-1.228-1.983h.931l.832 1.438h.036l.823-1.438Z" }) }),
    color: "bg-emerald-600"
  }
], je = ["image/jpeg", "image/jpg", "image/png", "image/gif", "image/tiff"], E0 = ({ file: a, index: o }) => {
  const [p, w] = He(null);
  G(() => {
    if (a) {
      const d = URL.createObjectURL(a);
      return w(d), () => URL.revokeObjectURL(d);
    }
  }, [a]);
  const b = Ue.filter((d) => d.type == a.type), s = S(R).state.componentState, { dispatch: m } = S(R), x = () => {
    m({
      type: "STORE_FILE_STATE",
      payload: {
        zoom: !0,
        fileSrc: URL.createObjectURL(a),
        index: o,
        isImage: je.includes(a.type),
        fileName: a.name,
        type: a.type,
        size: a.size
      }
    });
  };
  return /* @__PURE__ */ t.jsxs(t.Fragment, { children: [
    /* @__PURE__ */ t.jsx(
      "div",
      {
        "data-testid": "file-preview",
        onClick: () => x(),
        className: `${s.rounded && "rounded-lg"} border-solid border-slate-200 hover:shadow-lg  shadow-md hover:cursor-pointer`,
        children: je.includes(a.type) ? p && /* @__PURE__ */ t.jsx(
          "img",
          {
            "data-testid": "image-preview",
            src: p,
            className: `${s.fileHeight} ${s.fileWidth} object-fill scale-[0.9]`
          }
        ) : /* @__PURE__ */ t.jsx(
          "div",
          {
            "data-testid": "file-icon-preview",
            className: `${s.fileHeight} ${s.fileWidth} flex flex-col justify-center content-center items-center`,
            children: /* @__PURE__ */ t.jsx(
              "span",
              {
                className: `${b.length > 0 ? b[0].color : "bg-slate-400"} 
                        rounded flex w-16 justify-center h-20 items-center`,
                children: b.length > 0 ? b[0].icon : /* @__PURE__ */ t.jsxs(
                  "svg",
                  {
                    "data-testid": "default-icon",
                    xmlns: "http://www.w3.org/2000/svg",
                    className: "h-10 w-40 text-4xl",
                    viewBox: "0 0 16 16",
                    children: [
                      /* @__PURE__ */ t.jsx("path", { d: "M7 4.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0zm-.861 1.542 1.33.886 1.854-1.855a.25.25 0 0 1 .289-.047l1.888.974V7.5a.5.5 0 0 1-.5.5H5a.5.5 0 0 1-.5-.5V7s1.54-1.274 1.639-1.208zM5 9a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1H5z" }),
                      /* @__PURE__ */ t.jsx("path", { d: "M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1z" })
                    ]
                  }
                )
              }
            )
          }
        )
      }
    ),
    /* @__PURE__ */ t.jsx("div", { className: "h-5 w-44", children: p && /* @__PURE__ */ t.jsx(b0, { file: a, fileSrc: p, index: o, isImage: je.includes(a.type) }) })
  ] });
}, j0 = () => {
  const a = S(R).state.fileData, o = S(R).state.fileState;
  return /* @__PURE__ */ t.jsx("div", { id: "slider-count", className: "absolute bottom-5 right-0 left-0 flex items-center justify-center z-50", children: (o == null ? void 0 : o.index) !== null && /* @__PURE__ */ t.jsx("div", { className: "text-white text-1xl", children: `${o.index + 1} of ${a == null ? void 0 : a.length}` }) });
}, _0 = () => {
  const a = S(R).state.fileState, o = S(R).state.componentState, p = Ue.filter((d) => d.type === a.type), { dispatch: w } = S(R), b = () => {
    a.fileSrc && URL.revokeObjectURL(a.fileSrc), w({
      type: "STORE_FILE_STATE",
      payload: {
        zoom: !1,
        fileSrc: null,
        index: 0,
        isImage: !1,
        fileName: null,
        type: null,
        size: 0
      }
    });
  }, s = () => {
    w({ type: "GET_NEXT_FILE" });
  }, m = () => {
    w({ type: "GET_PREV_FILE" });
  }, x = () => {
    const d = document.documentElement;
    document.fullscreenElement ? document.exitFullscreen && document.exitFullscreen() : d.requestFullscreen && d.requestFullscreen();
  };
  return a.zoom ? /* @__PURE__ */ t.jsx("div", { children: /* @__PURE__ */ t.jsxs(
    "div",
    {
      id: "slider",
      className: "fixed top-0 left-0 right-0 bottom-0 flex items-center justify-center bg-black bg-opacity-80 z-20",
      children: [
        /* @__PURE__ */ t.jsxs(
          "div",
          {
            id: "slider-header",
            className: "absolute top-0 left-0 right-0 flex justify-between items-center p-2 shadow-md shadow-gray-950 bg-black bg-opacity-80",
            children: [
              /* @__PURE__ */ t.jsx("span", { className: "text-white flex-1 ml-14 max-sm:ml-1", children: a.fileName }),
              /* @__PURE__ */ t.jsx(
                "button",
                {
                  className: "text-white text-sm flex items-center mr-4  max-sm:mr-1 hover:bg-white hover:text-black rounded-lg pl-2 pr-2 pt-1 pb-1",
                  onClick: x,
                  "aria-label": "toggle-fullscreen",
                  children: /* @__PURE__ */ t.jsx(
                    "svg",
                    {
                      xmlns: "http://www.w3.org/2000/svg",
                      fill: "currentColor",
                      className: "h-4 stroke-2",
                      viewBox: "0 0 16 16",
                      children: /* @__PURE__ */ t.jsx("path", { d: "M1.5 1a.5.5 0 0 0-.5.5v4a.5.5 0 0 1-1 0v-4A1.5 1.5 0 0 1 1.5 0h4a.5.5 0 0 1 0 1h-4zM10 .5a.5.5 0 0 1 .5-.5h4A1.5 1.5 0 0 1 16 1.5v4a.5.5 0 0 1-1 0v-4a.5.5 0 0 0-.5-.5h-4a.5.5 0 0 1-.5-.5zM.5 10a.5.5 0 0 1 .5.5v4a.5.5 0 0 0 .5.5h4a.5.5 0 0 1 0 1h-4A1.5 1.5 0 0 1 0 14.5v-4a.5.5 0 0 1 .5-.5zm15 0a.5.5 0 0 1 .5.5v4a1.5 1.5 0 0 1-1.5 1.5h-4a.5.5 0 0 1 0-1h4a.5.5 0 0 0 .5-.5v-4a.5.5 0 0 1 .5-.5z" })
                    }
                  )
                }
              ),
              a.fileSrc && /* @__PURE__ */ t.jsxs(
                "a",
                {
                  href: a.fileSrc,
                  target: "_blank",
                  rel: "noreferrer",
                  download: !0,
                  className: "text-white text-sm flex items-center mr-4 max-sm:mr-1 hover:bg-white hover:text-black rounded-lg pl-2 pr-2 pt-1 pb-1",
                  children: [
                    /* @__PURE__ */ t.jsx(
                      "svg",
                      {
                        fill: "none",
                        stroke: "currentColor",
                        className: "mr-1 h-4 stroke-2",
                        viewBox: "0 0 24 24",
                        xmlns: "http://www.w3.org/2000/svg",
                        "aria-hidden": "true",
                        children: /* @__PURE__ */ t.jsx(
                          "path",
                          {
                            strokeLinecap: "round",
                            strokeLinejoin: "round",
                            d: "M3 16.5v2.25A2.25 2.25 0 005.25 21h13.5A2.25 2.25 0 0021 18.75V16.5M16.5 12L12 16.5m0 0L7.5 12m4.5 4.5V3"
                          }
                        )
                      }
                    ),
                    "Download"
                  ]
                }
              ),
              /* @__PURE__ */ t.jsxs(
                "button",
                {
                  className: "text-white text-sm flex items-center mr-2 max-sm:mr-0 hover:bg-white hover:text-black rounded-lg pl-2 pr-2 pt-1 pb-1",
                  onClick: b,
                  children: [
                    /* @__PURE__ */ t.jsx(
                      "svg",
                      {
                        xmlns: "http://www.w3.org/2000/svg",
                        fill: "none",
                        viewBox: "0 0 24 24",
                        strokeWidth: "1.5",
                        stroke: "currentColor",
                        className: "mr-1 w-4 h-4",
                        children: /* @__PURE__ */ t.jsx("path", { strokeLinecap: "round", strokeLinejoin: "round", d: "M6 18L18 6M6 6l12 12" })
                      }
                    ),
                    "Close"
                  ]
                }
              )
            ]
          }
        ),
        /* @__PURE__ */ t.jsx(
          "button",
          {
            id: "slider-prev-button",
            className: "absolute top-1/2 left-1 ml-2 p-1 max-sm:top-3/4 transform -translate-y-1/2 z-20 text-white rounded-full hover:bg-white hover:text-black",
            onClick: m,
            "data-testid": "prev-file",
            children: /* @__PURE__ */ t.jsx(
              "svg",
              {
                fill: "none",
                className: "h-10 w-10 stroke-white transform rotate-180 stroke-2 hover:stroke-black",
                viewBox: "0 0 24 24",
                xmlns: "http://www.w3.org/2000/svg",
                "aria-hidden": "true",
                children: /* @__PURE__ */ t.jsx(
                  "path",
                  {
                    strokeLinecap: "round",
                    strokeLinejoin: "round",
                    d: "M8.25 4.5l7.5 7.5-7.5 7.5"
                  }
                )
              }
            )
          }
        ),
        /* @__PURE__ */ t.jsx(
          "div",
          {
            id: "file-slider",
            "data-testid": "image-slider",
            className: `${o.rounded && "rounded-lg"} relative w-[130vh] max-sm:w-96 max-md:w-[75vh] h-[78vh] max-sm:h-52 max-md:h-80   bg-slate-400 overflow-hidden transition-all delay-750 ease-in`,
            children: a.isImage ? a.fileSrc && /* @__PURE__ */ t.jsx("img", { className: "object-fit w-full h-full ", src: a.fileSrc, alt: "Zoomed Image" }) : /* @__PURE__ */ t.jsx("span", { className: "flex w-full h-full items-center justify-center text-4xl", children: /* @__PURE__ */ t.jsx(
              "span",
              {
                className: `${p.length > 0 ? p[0].color : "bg-slate-400"} rounded flex justify-center  w-48 h-48 items-center`,
                children: p.length > 0 ? p[0].icon : /* @__PURE__ */ t.jsxs(
                  "svg",
                  {
                    "data-testid": "default-icon",
                    xmlns: "http://www.w3.org/2000/svg",
                    className: "h-10 w-40 text-4xl",
                    viewBox: "0 0 16 16",
                    children: [
                      /* @__PURE__ */ t.jsx("path", { d: "M7 4.25a.75.75 0 1 1-1.5 0 .75.75 0 0 1 1.5 0zm-.861 1.542 1.33.886 1.854-1.855a.25.25 0 0 1 .289-.047l1.888.974V7.5a.5.5 0 0 1-.5.5H5a.5.5 0 0 1-.5-.5V7s1.54-1.274 1.639-1.208zM5 9a.5.5 0 0 0 0 1h6a.5.5 0 0 0 0-1H5zm0 2a.5.5 0 0 0 0 1h3a.5.5 0 0 0 0-1H5z" }),
                      /* @__PURE__ */ t.jsx("path", { d: "M2 2a2 2 0 0 1 2-2h8a2 2 0 0 1 2 2v12a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm10-1H4a1 1 0 0 0-1 1v12a1 1 0 0 0 1 1h8a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1z" })
                    ]
                  }
                )
              }
            ) })
          }
        ),
        /* @__PURE__ */ t.jsx(
          "button",
          {
            id: "slider-next-button",
            className: "absolute top-1/2 mr-2 p-1 right-1 max-sm:top-3/4  transform -translate-y-1/2 text-white rounded-full hover:bg-white hover:text-black",
            onClick: s,
            "data-testid": "next-file",
            children: /* @__PURE__ */ t.jsx(
              "svg",
              {
                fill: "none",
                className: " stroke-white h-10 w-10 stroke-2 hover:stroke-black",
                viewBox: "0 0 24 24",
                xmlns: "http://www.w3.org/2000/svg",
                "aria-hidden": "true",
                children: /* @__PURE__ */ t.jsx(
                  "path",
                  {
                    strokeLinecap: "round",
                    strokeLinejoin: "round",
                    d: "M8.25 4.5l7.5 7.5-7.5 7.5"
                  }
                )
              }
            )
          }
        ),
        o.showSliderCount ? /* @__PURE__ */ t.jsx(j0, {}) : /* @__PURE__ */ t.jsx(t.Fragment, {})
      ]
    }
  ) }) : null;
}, R0 = (a) => /* @__PURE__ */ t.jsx(t.Fragment, { children: /* @__PURE__ */ t.jsx("div", { children: /* @__PURE__ */ t.jsxs("div", { className: "flex justify-between  bg-gray-200 dark:bg-gray-600 ", children: [
  /* @__PURE__ */ t.jsx("div", { className: "h-10 text-sm pt-2 ml-2 font-medium", children: /* @__PURE__ */ t.jsx("span", { className: "bg-gray-100 text-gray-800 text-sm font-medium mr-2 px-2.5 py-0.5 rounded dark:bg-gray-700 dark:text-gray-300", children: `Files: ${a.fileData.length}` }) }),
  /* @__PURE__ */ t.jsxs(
    "label",
    {
      htmlFor: "fileInput",
      className: `${a.disabled ? "cursor-not-allowed opacity-50  py-1 px-2 mt-1 mr-2 mb-1 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-full border border-gray-200 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700" : "cursor-pointer py-1 px-2 mt-1 mr-2 mb-1 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-full border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:bg-gray-800 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700"} `,
      children: [
        "+ Add more",
        /* @__PURE__ */ t.jsx(
          "input",
          {
            id: "fileInput",
            disabled: a.disabled,
            type: "file",
            onChange: (o) => {
              a.handleImage(o), a.onChange && a.onChange(o);
            },
            style: { display: "none" },
            multiple: a.multiple ?? !0,
            accept: a.accept ?? ""
          }
        )
      ]
    }
  )
] }) }) }), S0 = Re.lazy(async () => {
  const { ReactPhotoEditor: a } = await import("./react-photo-editor.es-282efa2c.js");
  return await import("./style-c6389719.js"), { default: a };
}), T0 = ({
  files: a,
  url: o,
  downloadFile: p,
  removeFile: w,
  showFileSize: b,
  showSliderCount: s,
  allowEditing: m,
  multiple: x,
  accept: d,
  maxFileSize: N,
  maxFiles: g,
  width: T,
  rounded: k,
  height: V,
  fileHeight: q,
  fileWidth: te,
  disabled: I,
  onChange: F,
  onRemove: j,
  onError: z,
  getFiles: re,
  onClick: ae,
  onDrop: ne
}) => {
  const A = S(R).state.fileData, pe = S(R).state.fileState, ie = S(R).state.componentState, M = S(R).state.imageEditorState, { dispatch: P } = S(R), U = (c) => {
    let v = !1;
    if (g && (A.length + c.length > g || c.length > g))
      throw v = !0, z && z(new Error(`Max ${g} files are allowed to be selected`)), new Error(`Max ${g} files are allowed to be selected`);
    return N && c.forEach((H) => {
      if (H.size > N)
        throw v = !0, z && z(new Error(`File size limit exceeded: ${H.name}`)), new Error(`File size limit exceeded: ${H.name}`);
    }), v;
  };
  G(() => {
    async function c() {
      try {
        if (o) {
          const v = [
            { type: "image/jpeg", ext: ".jpg" },
            { type: "image/jpg", ext: ".jpg" },
            { type: "image/png", ext: ".png" },
            { type: "image/gif", ext: "gif" },
            { type: "image/tiff", ext: ".tiff" }
          ], J = await (await fetch(o)).blob();
          let fe = null;
          const W = v.filter((Z) => Z.type === J.type);
          W.length > 0 && (fe = W[0].ext);
          const K = new File([J], "file" + fe, {
            type: J.type
          });
          P({ type: "STORE_FILE_DATA", payload: { files: [K] } });
        }
      } catch (v) {
        if (v instanceof Error)
          throw z && z(v), v;
      }
    }
    c();
  }, []), G(() => {
    a && a.length > 0 && (U(a) || P({ type: "STORE_FILE_DATA", payload: { files: a } }));
  }, [a]), G(() => {
    P({
      type: "SET_COMPONENT_STATE",
      payload: {
        downloadFile: p ?? !0,
        removeFile: w ?? !0,
        showFileSize: b ?? !0,
        showSliderCount: s ?? !0,
        rounded: k ?? !0,
        fileHeight: q ?? "h-32",
        fileWidth: te ?? "w-44",
        disabled: I ?? !1,
        allowEditing: m ?? !1
      }
    });
  }, [
    p,
    w,
    b,
    s,
    q,
    te,
    k,
    I,
    m
  ]);
  const C = (c) => {
    const v = Array.from(c.target.files || []);
    U(v) || P({ type: "APPEND_FILE_DATA", payload: { files: v } });
  }, D = (c) => {
    P({ type: "REMOVE_FILE_DATA", payload: c }), j && j(c);
  }, $ = (c) => {
    ae && ae(c);
  };
  G(() => {
    re && re(A);
  }, [A]);
  const le = (c) => {
    c.preventDefault(), c.dataTransfer.dropEffect = "copy";
  }, oe = (c) => {
    c.preventDefault();
  }, se = (c) => {
    c.preventDefault();
    const v = Array.from(c.dataTransfer.files);
    v && v.length > 0 && (U(v) || (P({ type: "APPEND_FILE_DATA", payload: { files: v } }), ne && ne(c)));
  }, ce = (c) => {
    if (c && M.index != null) {
      const v = Array.from(A);
      v[M.index] = c, P({
        type: "STORE_FILE_DATA",
        payload: {
          files: v
        }
      });
    }
  }, ue = () => {
    P({
      type: "SET_IMAGE_EDITOR_DATA",
      payload: {
        isEditing: !1,
        file: null,
        index: null
      }
    });
  };
  return pe.zoom ? /* @__PURE__ */ t.jsx("div", { children: /* @__PURE__ */ t.jsx(_0, {}) }) : M.isEditing && M.file ? /* @__PURE__ */ t.jsx(m0, { fallback: /* @__PURE__ */ t.jsx(t.Fragment, {}), children: /* @__PURE__ */ t.jsx(
    S0,
    {
      open: M.isEditing,
      file: M.file,
      onSaveImage: ce,
      onClose: ue,
      downloadOnSave: !1
    }
  ) }) : /* @__PURE__ */ t.jsx("div", { className: "w-full", children: /* @__PURE__ */ t.jsx("div", { className: "flex flex-row max-h-2", children: /* @__PURE__ */ t.jsxs("div", { className: `${T ?? "basis-11/12"} mx-auto`, children: [
    A.length > 0 && /* @__PURE__ */ t.jsx(
      R0,
      {
        fileData: A,
        multiple: x,
        disabled: I,
        accept: d,
        onChange: F,
        handleImage: C
      }
    ),
    /* @__PURE__ */ t.jsx(
      "div",
      {
        className: `${V && `overflow-auto ${V}`} ${A.length == 0 && `border-2 border-dashed border-gray-300 ${I ? "" : "hover:bg-stone-200"} `} flex flex-row flex-wrap gap-4 p-6 bg-stone-100  shadow dark:bg-gray-800 `,
        onDragOver: le,
        onDragLeave: oe,
        onDrop: se,
        "data-testid": "dropzone",
        children: A.length > 0 ? A.map((c, v) => /* @__PURE__ */ t.jsxs("div", { className: "relative pb-5 group", onClick: () => $(c), children: [
          /* @__PURE__ */ t.jsx("div", { className: "ml-9", children: ie.removeFile ? /* @__PURE__ */ t.jsx(
            "svg",
            {
              xmlns: "http://www.w3.org/2000/svg",
              "data-testid": "remove-file-button",
              onClick: () => D(c),
              className: "absolute -top-2 right-0 z-10 text-black opacity-0 group-hover:opacity-100 transition-opacity cursor-pointer h-5 w-5",
              fill: "currentColor",
              viewBox: "0 0 16 16",
              children: /* @__PURE__ */ t.jsx("path", { d: "M16 8A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM5.354 4.646a.5.5 0 1 0-.708.708L7.293 8l-2.647 2.646a.5.5 0 0 0 .708.708L8 8.707l2.646 2.647a.5.5 0 0 0 .708-.708L8.707 8l2.647-2.646a.5.5 0 0 0-.708-.708L8 7.293 5.354 4.646z" })
            }
          ) : /* @__PURE__ */ t.jsx(t.Fragment, {}) }),
          /* @__PURE__ */ t.jsx("div", { className: "clear-right", children: /* @__PURE__ */ t.jsx(E0, { file: c, index: v }) })
        ] }, v)) : /* @__PURE__ */ t.jsxs(
          "label",
          {
            htmlFor: "fileInput",
            className: `text-black dark:text-slate-100 ${I ? "mx-auto cursor-not-allowed  flex items-center" : "mx-auto cursor-pointer  flex items-center"}`,
            children: [
              "Drop files here, or click to browse files",
              /* @__PURE__ */ t.jsx(
                "input",
                {
                  id: "fileInput",
                  disabled: I,
                  type: "file",
                  onChange: (c) => {
                    C(c), F && F(c);
                  },
                  multiple: x ?? !0,
                  accept: d ?? "",
                  style: { display: "none" }
                }
              )
            ]
          }
        )
      }
    )
  ] }) }) });
};
const C0 = (a) => /* @__PURE__ */ t.jsx(y0, { children: /* @__PURE__ */ t.jsx(T0, { ...a }) });
export {
  C0 as ReactFilesPreview
};
