// react-files-preview/dist/react-photo-editor.es-282efa2c.js
import ze, { useRef as ur, useState as _, useEffect as Fe } from "react";
var we = { exports: {} }, oe = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var De;
function fr() {
  if (De)
    return oe;
  De = 1;
  var R = ze, P = Symbol.for("react.element"), X = Symbol.for("react.fragment"), E = Object.prototype.hasOwnProperty, M = R.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, U = { key: !0, ref: !0, __self: !0, __source: !0 };
  function T(w, p, j) {
    var y, b = {}, k = null, z = null;
    j !== void 0 && (k = "" + j), p.key !== void 0 && (k = "" + p.key), p.ref !== void 0 && (z = p.ref);
    for (y in p)
      E.call(p, y) && !U.hasOwnProperty(y) && (b[y] = p[y]);
    if (w && w.defaultProps)
      for (y in p = w.defaultProps, p)
        b[y] === void 0 && (b[y] = p[y]);
    return { $$typeof: P, type: w, key: k, ref: z, props: b, _owner: M.current };
  }
  return oe.Fragment = X, oe.jsx = T, oe.jsxs = T, oe;
}
var ye = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Me;
function pr() {
  return Me || (Me = 1, process.env.NODE_ENV !== "production" && function() {
    var R = ze, P = Symbol.for("react.element"), X = Symbol.for("react.portal"), E = Symbol.for("react.fragment"), M = Symbol.for("react.strict_mode"), U = Symbol.for("react.profiler"), T = Symbol.for("react.provider"), w = Symbol.for("react.context"), p = Symbol.for("react.forward_ref"), j = Symbol.for("react.suspense"), y = Symbol.for("react.suspense_list"), b = Symbol.for("react.memo"), k = Symbol.for("react.lazy"), z = Symbol.for("react.offscreen"), I = Symbol.iterator, Z = "@@iterator";
    function W(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = I && e[I] || e[Z];
      return typeof r == "function" ? r : null;
    }
    var N = R.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function d(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), a = 1; a < r; a++)
          t[a - 1] = arguments[a];
        K("error", e, t);
      }
    }
    function K(e, r, t) {
      {
        var a = N.ReactDebugCurrentFrame, l = a.getStackAddendum();
        l !== "" && (r += "%s", t = t.concat([l]));
        var c = t.map(function(i) {
          return String(i);
        });
        c.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, c);
      }
    }
    var B = !1, Q = !1, L = !1, se = !1, G = !1, ee;
    ee = Symbol.for("react.module.reference");
    function re(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === E || e === U || G || e === M || e === j || e === y || se || e === z || B || Q || L || typeof e == "object" && e !== null && (e.$$typeof === k || e.$$typeof === b || e.$$typeof === T || e.$$typeof === w || e.$$typeof === p || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === ee || e.getModuleId !== void 0));
    }
    function ie(e, r, t) {
      var a = e.displayName;
      if (a)
        return a;
      var l = r.displayName || r.name || "";
      return l !== "" ? t + "(" + l + ")" : t;
    }
    function $(e) {
      return e.displayName || "Context";
    }
    function g(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && d("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case E:
          return "Fragment";
        case X:
          return "Portal";
        case U:
          return "Profiler";
        case M:
          return "StrictMode";
        case j:
          return "Suspense";
        case y:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case w:
            var r = e;
            return $(r) + ".Consumer";
          case T:
            var t = e;
            return $(t._context) + ".Provider";
          case p:
            return ie(e, e.render, "ForwardRef");
          case b:
            var a = e.displayName || null;
            return a !== null ? a : g(e.type) || "Memo";
          case k: {
            var l = e, c = l._payload, i = l._init;
            try {
              return g(i(c));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var O = Object.assign, F = 0, te, le, ce, A, ue, fe, o;
    function u() {
    }
    u.__reactDisabledLog = !0;
    function v() {
      {
        if (F === 0) {
          te = console.log, le = console.info, ce = console.warn, A = console.error, ue = console.group, fe = console.groupCollapsed, o = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: u,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        F++;
      }
    }
    function H() {
      {
        if (F--, F === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: O({}, e, {
              value: te
            }),
            info: O({}, e, {
              value: le
            }),
            warn: O({}, e, {
              value: ce
            }),
            error: O({}, e, {
              value: A
            }),
            group: O({}, e, {
              value: ue
            }),
            groupCollapsed: O({}, e, {
              value: fe
            }),
            groupEnd: O({}, e, {
              value: o
            })
          });
        }
        F < 0 && d("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var S = N.ReactCurrentDispatcher, ne;
    function Y(e, r, t) {
      {
        if (ne === void 0)
          try {
            throw Error();
          } catch (l) {
            var a = l.stack.trim().match(/\n( *(at )?)/);
            ne = a && a[1] || "";
          }
        return `
` + ne + e;
      }
    }
    var V = !1, D;
    {
      var We = typeof WeakMap == "function" ? WeakMap : Map;
      D = new We();
    }
    function je(e, r) {
      if (!e || V)
        return "";
      {
        var t = D.get(e);
        if (t !== void 0)
          return t;
      }
      var a;
      V = !0;
      var l = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var c;
      c = S.current, S.current = null, v();
      try {
        if (r) {
          var i = function() {
            throw Error();
          };
          if (Object.defineProperty(i.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(i, []);
            } catch (C) {
              a = C;
            }
            Reflect.construct(e, [], i);
          } else {
            try {
              i.call();
            } catch (C) {
              a = C;
            }
            e.call(i.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (C) {
            a = C;
          }
          e();
        }
      } catch (C) {
        if (C && a && typeof C.stack == "string") {
          for (var s = C.stack.split(`
`), h = a.stack.split(`
`), f = s.length - 1, m = h.length - 1; f >= 1 && m >= 0 && s[f] !== h[m]; )
            m--;
          for (; f >= 1 && m >= 0; f--, m--)
            if (s[f] !== h[m]) {
              if (f !== 1 || m !== 1)
                do
                  if (f--, m--, m < 0 || s[f] !== h[m]) {
                    var x = `
` + s[f].replace(" at new ", " at ");
                    return e.displayName && x.includes("<anonymous>") && (x = x.replace("<anonymous>", e.displayName)), typeof e == "function" && D.set(e, x), x;
                  }
                while (f >= 1 && m >= 0);
              break;
            }
        }
      } finally {
        V = !1, S.current = c, H(), Error.prepareStackTrace = l;
      }
      var J = e ? e.displayName || e.name : "", Le = J ? Y(J) : "";
      return typeof e == "function" && D.set(e, Le), Le;
    }
    function Be(e, r, t) {
      return je(e, !1);
    }
    function Ae(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function pe(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return je(e, Ae(e));
      if (typeof e == "string")
        return Y(e);
      switch (e) {
        case j:
          return Y("Suspense");
        case y:
          return Y("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case p:
            return Be(e.render);
          case b:
            return pe(e.type, r, t);
          case k: {
            var a = e, l = a._payload, c = a._init;
            try {
              return pe(c(l), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var de = Object.prototype.hasOwnProperty, ke = {}, Ne = N.ReactDebugCurrentFrame;
    function me(e) {
      if (e) {
        var r = e._owner, t = pe(e.type, e._source, r ? r.type : null);
        Ne.setExtraStackFrame(t);
      } else
        Ne.setExtraStackFrame(null);
    }
    function He(e, r, t, a, l) {
      {
        var c = Function.call.bind(de);
        for (var i in e)
          if (c(e, i)) {
            var s = void 0;
            try {
              if (typeof e[i] != "function") {
                var h = Error((a || "React class") + ": " + t + " type `" + i + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[i] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw h.name = "Invariant Violation", h;
              }
              s = e[i](r, i, a, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (f) {
              s = f;
            }
            s && !(s instanceof Error) && (me(l), d("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", a || "React class", t, i, typeof s), me(null)), s instanceof Error && !(s.message in ke) && (ke[s.message] = !0, me(l), d("Failed %s type: %s", t, s.message), me(null));
          }
      }
    }
    var Ye = Array.isArray;
    function he(e) {
      return Ye(e);
    }
    function Ve(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function qe(e) {
      try {
        return _e(e), !1;
      } catch {
        return !0;
      }
    }
    function _e(e) {
      return "" + e;
    }
    function Re(e) {
      if (qe(e))
        return d("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ve(e)), _e(e);
    }
    var ae = N.ReactCurrentOwner, Je = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Oe, Se, ge;
    ge = {};
    function Xe(e) {
      if (de.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Ze(e) {
      if (de.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function Ke(e, r) {
      if (typeof e.ref == "string" && ae.current && r && ae.current.stateNode !== r) {
        var t = g(ae.current.type);
        ge[t] || (d('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', g(ae.current.type), e.ref), ge[t] = !0);
      }
    }
    function Qe(e, r) {
      {
        var t = function() {
          Oe || (Oe = !0, d("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function Ge(e, r) {
      {
        var t = function() {
          Se || (Se = !0, d("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var er = function(e, r, t, a, l, c, i) {
      var s = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: P,
        // Built-in properties that belong on the element
        type: e,
        key: r,
        ref: t,
        props: i,
        // Record the component responsible for creating this element.
        _owner: c
      };
      return s._store = {}, Object.defineProperty(s._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(s, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: a
      }), Object.defineProperty(s, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: l
      }), Object.freeze && (Object.freeze(s.props), Object.freeze(s)), s;
    };
    function rr(e, r, t, a, l) {
      {
        var c, i = {}, s = null, h = null;
        t !== void 0 && (Re(t), s = "" + t), Ze(r) && (Re(r.key), s = "" + r.key), Xe(r) && (h = r.ref, Ke(r, l));
        for (c in r)
          de.call(r, c) && !Je.hasOwnProperty(c) && (i[c] = r[c]);
        if (e && e.defaultProps) {
          var f = e.defaultProps;
          for (c in f)
            i[c] === void 0 && (i[c] = f[c]);
        }
        if (s || h) {
          var m = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          s && Qe(i, m), h && Ge(i, m);
        }
        return er(e, s, h, l, a, ae.current, i);
      }
    }
    var ve = N.ReactCurrentOwner, Ce = N.ReactDebugCurrentFrame;
    function q(e) {
      if (e) {
        var r = e._owner, t = pe(e.type, e._source, r ? r.type : null);
        Ce.setExtraStackFrame(t);
      } else
        Ce.setExtraStackFrame(null);
    }
    var xe;
    xe = !1;
    function be(e) {
      return typeof e == "object" && e !== null && e.$$typeof === P;
    }
    function Ee() {
      {
        if (ve.current) {
          var e = g(ve.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function tr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var $e = {};
    function nr(e) {
      {
        var r = Ee();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function Pe(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = nr(r);
        if ($e[t])
          return;
        $e[t] = !0;
        var a = "";
        e && e._owner && e._owner !== ve.current && (a = " It was passed a child from " + g(e._owner.type) + "."), q(e), d('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, a), q(null);
      }
    }
    function Te(e, r) {
      {
        if (typeof e != "object")
          return;
        if (he(e))
          for (var t = 0; t < e.length; t++) {
            var a = e[t];
            be(a) && Pe(a, r);
          }
        else if (be(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var l = W(e);
          if (typeof l == "function" && l !== e.entries)
            for (var c = l.call(e), i; !(i = c.next()).done; )
              be(i.value) && Pe(i.value, r);
        }
      }
    }
    function ar(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === p || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        r.$$typeof === b))
          t = r.propTypes;
        else
          return;
        if (t) {
          var a = g(r);
          He(t, e.props, "prop", a, e);
        } else if (r.PropTypes !== void 0 && !xe) {
          xe = !0;
          var l = g(r);
          d("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", l || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && d("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function or(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var a = r[t];
          if (a !== "children" && a !== "key") {
            q(e), d("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", a), q(null);
            break;
          }
        }
        e.ref !== null && (q(e), d("Invalid attribute `ref` supplied to `React.Fragment`."), q(null));
      }
    }
    function Ie(e, r, t, a, l, c) {
      {
        var i = re(e);
        if (!i) {
          var s = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (s += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var h = tr(l);
          h ? s += h : s += Ee();
          var f;
          e === null ? f = "null" : he(e) ? f = "array" : e !== void 0 && e.$$typeof === P ? (f = "<" + (g(e.type) || "Unknown") + " />", s = " Did you accidentally export a JSX literal instead of a component?") : f = typeof e, d("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", f, s);
        }
        var m = rr(e, r, t, l, c);
        if (m == null)
          return m;
        if (i) {
          var x = r.children;
          if (x !== void 0)
            if (a)
              if (he(x)) {
                for (var J = 0; J < x.length; J++)
                  Te(x[J], e);
                Object.freeze && Object.freeze(x);
              } else
                d("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Te(x, e);
        }
        return e === E ? or(m) : ar(m), m;
      }
    }
    function sr(e, r, t) {
      return Ie(e, r, t, !0);
    }
    function ir(e, r, t) {
      return Ie(e, r, t, !1);
    }
    var lr = ir, cr = sr;
    ye.Fragment = E, ye.jsx = lr, ye.jsxs = cr;
  }()), ye;
}
process.env.NODE_ENV === "production" ? we.exports = fr() : we.exports = pr();
var n = we.exports;
const Ue = "ml-2 text-md outline-none py-1 px-2 text-sm font-medium text-gray-900 focus:outline-none bg-white rounded-lg border border-gray-200 hover:bg-gray-100 hover:text-blue-700 focus:z-10 focus:ring-4 focus:ring-gray-200 dark:focus:ring-gray-700 dark:text-gray-400 dark:border-gray-600 dark:hover:text-white dark:hover:bg-gray-700", mr = ({
  file: R,
  onSaveImage: P,
  allowColorEditing: X = !0,
  allowFlip: E = !0,
  allowRotate: M = !0,
  allowZoom: U = !0,
  downloadOnSave: T,
  open: w,
  onClose: p
}) => {
  const j = ur(null), [y, b] = _(""), [k, z] = _(""), [I, Z] = _(100), [W, N] = _(100), [d, K] = _(100), [B, Q] = _(0), [L, se] = _(0), [G, ee] = _(!1), [re, ie] = _(!1), [$, g] = _(1);
  Fe(() => {
    if (R) {
      const o = URL.createObjectURL(R);
      return b(o), z(R.name), () => {
        URL.revokeObjectURL(o);
      };
    }
  }, [R, w]), Fe(() => {
    O();
  }, [R, y, L, G, re, $, I, W, d, B]);
  const O = () => {
    const o = j.current, u = o == null ? void 0 : o.getContext("2d"), v = new Image();
    v.src = y, v.onload = () => {
      if (o && u) {
        const H = v.width * $, S = v.height * $, ne = (v.width - H) / 2, Y = (v.height - S) / 2;
        if (o.width = v.width, o.height = v.height, u.filter = F(), u.save(), L) {
          const V = o.width / 2, D = o.height / 2;
          u.translate(V, D), u.rotate(L * Math.PI / 180), u.translate(-V, -D);
        }
        G && (u.translate(o.width, 0), u.scale(-1, 1)), re && (u.translate(0, o.height), u.scale(1, -1)), u.translate(ne, Y), u.scale($, $), u.drawImage(v, 0, 0, o.width, o.height), u.restore();
      }
    };
  }, F = () => `brightness(${I}%) contrast(${W}%) grayscale(${B}%) saturate(${d}%)`, te = (o) => {
    var u;
    se(parseInt((u = o == null ? void 0 : o.target) == null ? void 0 : u.value));
  }, le = () => {
    g((o) => o + 0.1);
  }, ce = () => {
    g((o) => Math.max(o - 0.1, 0.1));
  }, A = () => {
    Z(100), N(100), K(100), Q(0), se(0), ee(!1), ie(!1), g(1);
  }, ue = () => {
    const o = j.current;
    o && (o.toBlob((u) => {
      if (u) {
        const v = new File([u], k, { type: u.type });
        if (T) {
          const H = URL.createObjectURL(u), S = document.createElement("a");
          S.download = `${k}`, S.href = H, S.click(), URL.revokeObjectURL(H);
        }
        P(v), p && p();
      }
    }), A());
  }, fe = () => {
    A(), p && p();
  };
  return /* @__PURE__ */ n.jsx(n.Fragment, { children: w && /* @__PURE__ */ n.jsxs(n.Fragment, { children: [
    /* @__PURE__ */ n.jsx("div", { "data-testid": "photo-editor-main", className: "photo-editor-main justify-center items-center flex overflow-auto fixed inset-0 z-50", children: /* @__PURE__ */ n.jsxs("div", { className: "relative rounded-lg shadow-lg w-[40rem] max-sm:w-[22rem] bg-white h-[38rem] dark:bg-slate-800", children: [
      /* @__PURE__ */ n.jsxs("div", { className: "flex justify-end p-2 rounded-t", children: [
        /* @__PURE__ */ n.jsx(
          "button",
          {
            className: Ue,
            onClick: fe,
            children: "Close"
          }
        ),
        /* @__PURE__ */ n.jsx(
          "button",
          {
            className: Ue,
            onClick: () => ue(),
            "data-testid": "save-button",
            children: "Save"
          }
        )
      ] }),
      /* @__PURE__ */ n.jsx("div", { className: "p-4", children: /* @__PURE__ */ n.jsxs("div", { className: "flex flex-col", children: [
        /* @__PURE__ */ n.jsx("canvas", { className: "canvas max-w-lg max-h-[22rem] w-full object-fit mx-auto", "data-testid": "image-editor-canvas", id: "canvas", ref: j }),
        /* @__PURE__ */ n.jsx("div", { className: "items-center flex m-1 flex-col", children: /* @__PURE__ */ n.jsxs("div", { className: "flex flex-col bottom-12 gap-1 mt-4 max-sm:w-72 w-11/12 absolute ", children: [
          M && /* @__PURE__ */ n.jsxs("div", { className: "flex flex-row items-center", children: [
            /* @__PURE__ */ n.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Rotate: " }),
            /* @__PURE__ */ n.jsx(
              "input",
              {
                type: "range",
                value: L,
                step: "1",
                onChange: te,
                min: -180,
                max: 180,
                className: "ml-[1.8rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
              }
            ),
            /* @__PURE__ */ n.jsx(
              "input",
              {
                type: "number",
                value: L,
                onChange: te,
                className: "w-14 pl-1 rounded-md text-right"
              }
            )
          ] }),
          X && /* @__PURE__ */ n.jsxs(n.Fragment, { children: [
            /* @__PURE__ */ n.jsxs("div", { className: "flex flex-row justify-between items-center", "data-testid": "color-editing", children: [
              /* @__PURE__ */ n.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Brightness: " }),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  id: "default-range",
                  type: "range",
                  step: "1",
                  max: 200,
                  value: I,
                  onChange: (o) => Z(parseInt(o.target.value)),
                  className: "ml-2 w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  type: "number",
                  value: I,
                  onChange: (o) => Z(parseInt(o.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] }),
            /* @__PURE__ */ n.jsxs("div", { className: "flex items-center", children: [
              /* @__PURE__ */ n.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Contrast: " }),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  id: "default-range",
                  type: "range",
                  step: "1",
                  max: 200,
                  value: W,
                  onChange: (o) => N(parseInt(o.target.value)),
                  className: "ml-[1.1rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  type: "number",
                  value: W,
                  onChange: (o) => N(parseInt(o.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] }),
            /* @__PURE__ */ n.jsxs("div", { className: "flex items-center", children: [
              /* @__PURE__ */ n.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Saturate: " }),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  id: "default-range",
                  type: "range",
                  step: "1",
                  max: 200,
                  value: d,
                  onChange: (o) => K(parseInt(o.target.value)),
                  className: "ml-[1.2rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  type: "number",
                  value: d,
                  onChange: (o) => K(parseInt(o.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] }),
            /* @__PURE__ */ n.jsxs("div", { className: "flex items-center", children: [
              /* @__PURE__ */ n.jsx("label", { className: "text-xs font-medium text-gray-900 dark:text-white", children: "Grayscale: " }),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  id: "grayscaleSlider",
                  type: "range",
                  value: B,
                  max: "100",
                  onChange: (o) => Q(parseInt(o.target.value)),
                  className: "ml-[0.8rem] w-full h-1 bg-gray-200 rounded-lg appearance-none cursor-pointer range-sm dark:bg-gray-700"
                }
              ),
              /* @__PURE__ */ n.jsx(
                "input",
                {
                  type: "number",
                  value: B,
                  onChange: (o) => Q(parseInt(o.target.value)),
                  className: "w-14 pl-1 rounded-md text-right"
                }
              )
            ] })
          ] })
        ] }) }),
        /* @__PURE__ */ n.jsx("div", { className: "flex justify-center", children: /* @__PURE__ */ n.jsxs("div", { className: "mb-1 absolute bottom-0 mt-2", children: [
          /* @__PURE__ */ n.jsx("button", { className: "mx-1  focus:ring-4 rounded-md p-1", onClick: A, children: /* @__PURE__ */ n.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", "stroke-width": "2", "stroke-linecap": "round", "stroke-linejoin": "round", className: "lucide lucide-rotate-ccw dark:stroke-slate-200", children: [
            /* @__PURE__ */ n.jsx("path", { d: "M3 12a9 9 0 1 0 9-9 9.75 9.75 0 0 0-6.74 2.74L3 8" }),
            /* @__PURE__ */ n.jsx("path", { d: "M3 3v5h5" })
          ] }) }),
          E && /* @__PURE__ */ n.jsxs("div", { className: "inline-block", "data-testid": "flip-btns", children: [
            /* @__PURE__ */ n.jsx("button", { className: "mx-1 focus:ring-4 rounded-md p-1", onClick: () => ee(!G), children: /* @__PURE__ */ n.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-flip-horizontal dark:stroke-slate-200", children: [
              /* @__PURE__ */ n.jsx("path", { d: "M8 3H5a2 2 0 0 0-2 2v14c0 1.1.9 2 2 2h3" }),
              /* @__PURE__ */ n.jsx("path", { d: "M16 3h3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2h-3" }),
              /* @__PURE__ */ n.jsx("path", { d: "M12 20v2" }),
              /* @__PURE__ */ n.jsx("path", { d: "M12 14v2" }),
              /* @__PURE__ */ n.jsx("path", { d: "M12 8v2" }),
              /* @__PURE__ */ n.jsx("path", { d: "M12 2v2" })
            ] }) }),
            /* @__PURE__ */ n.jsx("button", { className: "mx-1 focus:ring-4 rounded-md p-1", onClick: () => ie(!re), children: /* @__PURE__ */ n.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-flip-vertical dark:stroke-slate-200", children: [
              /* @__PURE__ */ n.jsx("path", { d: "M21 8V5a2 2 0 0 0-2-2H5a2 2 0 0 0-2 2v3" }),
              /* @__PURE__ */ n.jsx("path", { d: "M21 16v3a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-3" }),
              /* @__PURE__ */ n.jsx("path", { d: "M4 12H2" }),
              /* @__PURE__ */ n.jsx("path", { d: "M10 12H8" }),
              /* @__PURE__ */ n.jsx("path", { d: "M16 12h-2" }),
              /* @__PURE__ */ n.jsx("path", { d: "M22 12h-2" })
            ] }) })
          ] }),
          U && /* @__PURE__ */ n.jsxs("div", { className: "inline-block", "data-testid": "zoom-btns", children: [
            /* @__PURE__ */ n.jsx("button", { "data-testid": "zoom-in", className: "mx-1 focus:ring-4 rounded-md p-1", onClick: le, children: /* @__PURE__ */ n.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-zoom-in dark:stroke-slate-200", children: [
              /* @__PURE__ */ n.jsx("circle", { cx: "11", cy: "11", r: "8" }),
              /* @__PURE__ */ n.jsx("line", { x1: "21", x2: "16.65", y1: "21", y2: "16.65" }),
              /* @__PURE__ */ n.jsx("line", { x1: "11", x2: "11", y1: "8", y2: "14" }),
              /* @__PURE__ */ n.jsx("line", { x1: "8", x2: "14", y1: "11", y2: "11" })
            ] }) }),
            /* @__PURE__ */ n.jsx("button", { "data-testid": "zoom-out", className: "mx-1 focus:ring-4 rounded-md p-1", onClick: ce, children: /* @__PURE__ */ n.jsxs("svg", { xmlns: "http://www.w3.org/2000/svg", width: "24", height: "24", viewBox: "0 0 24 24", fill: "none", stroke: "currentColor", strokeWidth: "2", strokeLinecap: "round", strokeLinejoin: "round", className: "lucide lucide-zoom-out dark:stroke-slate-200", children: [
              /* @__PURE__ */ n.jsx("circle", { cx: "11", cy: "11", r: "8" }),
              /* @__PURE__ */ n.jsx("line", { x1: "21", x2: "16.65", y1: "21", y2: "16.65" }),
              /* @__PURE__ */ n.jsx("line", { x1: "8", x2: "14", y1: "11", y2: "11" })
            ] }) })
          ] })
        ] }) })
      ] }) })
    ] }) }),
    /* @__PURE__ */ n.jsx("div", { className: "opacity-75 fixed inset-0 z-40 bg-black" })
  ] }) });
};
export {
  mr as ReactPhotoEditor
};
